// @formance/formance-sdk/sdk/models/errors/v2error.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.V2Error$ = exports.SchemasErrorCode$ = exports.V2Error = exports.SchemasErrorCode = void 0;
const z = __importStar(require("zod"));
var SchemasErrorCode;
(function (SchemasErrorCode) {
    SchemasErrorCode["Validation"] = "VALIDATION";
    SchemasErrorCode["NotFound"] = "NOT_FOUND";
    SchemasErrorCode["Internal"] = "INTERNAL";
})(SchemasErrorCode || (exports.SchemasErrorCode = SchemasErrorCode = {}));
/**
 * General error
 */
class V2Error extends Error {
    constructor(err) {
        super("");
        this.data$ = err;
        this.errorCode = err.errorCode;
        this.errorMessage = err.errorMessage;
        this.message =
            "message" in err && typeof err.message === "string"
                ? err.message
                : "API error occurred";
        this.name = "V2Error";
    }
}
exports.V2Error = V2Error;
/** @internal */
var SchemasErrorCode$;
(function (SchemasErrorCode$) {
    SchemasErrorCode$.inboundSchema = z.nativeEnum(SchemasErrorCode);
    SchemasErrorCode$.outboundSchema = SchemasErrorCode$.inboundSchema;
})(SchemasErrorCode$ || (exports.SchemasErrorCode$ = SchemasErrorCode$ = {}));
/** @internal */
var V2Error$;
(function (V2Error$) {
    V2Error$.inboundSchema = z
        .object({
        errorCode: SchemasErrorCode$.inboundSchema,
        errorMessage: z.string(),
    })
        .transform((v) => {
        return new V2Error(v);
    });
    V2Error$.outboundSchema = z
        .instanceof(V2Error)
        .transform((v) => v.data$)
        .pipe(z.object({
        errorCode: SchemasErrorCode$.outboundSchema,
        errorMessage: z.string(),
    }));
})(V2Error$ || (exports.V2Error$ = V2Error$ = {}));
