// @formance/formance-sdk/sdk/models/errors/walletserrorresponse.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WalletsErrorResponse$ = exports.SchemasWalletsErrorResponseErrorCode$ = exports.WalletsErrorResponse = exports.SchemasWalletsErrorResponseErrorCode = void 0;
const z = __importStar(require("zod"));
var SchemasWalletsErrorResponseErrorCode;
(function (SchemasWalletsErrorResponseErrorCode) {
    SchemasWalletsErrorResponseErrorCode["Validation"] = "VALIDATION";
    SchemasWalletsErrorResponseErrorCode["InternalError"] = "INTERNAL_ERROR";
    SchemasWalletsErrorResponseErrorCode["InsufficientFund"] = "INSUFFICIENT_FUND";
    SchemasWalletsErrorResponseErrorCode["HoldClosed"] = "HOLD_CLOSED";
})(SchemasWalletsErrorResponseErrorCode || (exports.SchemasWalletsErrorResponseErrorCode = SchemasWalletsErrorResponseErrorCode = {}));
/**
 * Error
 */
class WalletsErrorResponse extends Error {
    constructor(err) {
        super("");
        this.data$ = err;
        this.errorCode = err.errorCode;
        this.errorMessage = err.errorMessage;
        this.message =
            "message" in err && typeof err.message === "string"
                ? err.message
                : "API error occurred";
        this.name = "WalletsErrorResponse";
    }
}
exports.WalletsErrorResponse = WalletsErrorResponse;
/** @internal */
var SchemasWalletsErrorResponseErrorCode$;
(function (SchemasWalletsErrorResponseErrorCode$) {
    SchemasWalletsErrorResponseErrorCode$.inboundSchema = z.nativeEnum(SchemasWalletsErrorResponseErrorCode);
    SchemasWalletsErrorResponseErrorCode$.outboundSchema = SchemasWalletsErrorResponseErrorCode$.inboundSchema;
})(SchemasWalletsErrorResponseErrorCode$ || (exports.SchemasWalletsErrorResponseErrorCode$ = SchemasWalletsErrorResponseErrorCode$ = {}));
/** @internal */
var WalletsErrorResponse$;
(function (WalletsErrorResponse$) {
    WalletsErrorResponse$.inboundSchema = z
        .object({
        errorCode: SchemasWalletsErrorResponseErrorCode$.inboundSchema,
        errorMessage: z.string(),
    })
        .transform((v) => {
        return new WalletsErrorResponse(v);
    });
    WalletsErrorResponse$.outboundSchema = z
        .instanceof(WalletsErrorResponse)
        .transform((v) => v.data$)
        .pipe(z.object({
        errorCode: SchemasWalletsErrorResponseErrorCode$.outboundSchema,
        errorMessage: z.string(),
    }));
})(WalletsErrorResponse$ || (exports.WalletsErrorResponse$ = WalletsErrorResponse$ = {}));
