// @formance/formance-sdk/sdk/models/shared/accountscursor.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountsCursor$ = exports.Cursor$ = void 0;
const paymentsaccount_1 = require("./paymentsaccount");
const z = __importStar(require("zod"));
/** @internal */
var Cursor$;
(function (Cursor$) {
    Cursor$.inboundSchema = z.object({
        data: z.array(paymentsaccount_1.PaymentsAccount$.inboundSchema),
        hasMore: z.boolean(),
        next: z.string().optional(),
        pageSize: z.number().int(),
        previous: z.string().optional(),
    });
    Cursor$.outboundSchema = z.object({
        data: z.array(paymentsaccount_1.PaymentsAccount$.outboundSchema),
        hasMore: z.boolean(),
        next: z.string().optional(),
        pageSize: z.number().int(),
        previous: z.string().optional(),
    });
})(Cursor$ || (exports.Cursor$ = Cursor$ = {}));
/** @internal */
var AccountsCursor$;
(function (AccountsCursor$) {
    AccountsCursor$.inboundSchema = z.object({
        cursor: z.lazy(() => Cursor$.inboundSchema),
    });
    AccountsCursor$.outboundSchema = z.object({
        cursor: z.lazy(() => Cursor$.outboundSchema),
    });
})(AccountsCursor$ || (exports.AccountsCursor$ = AccountsCursor$ = {}));
