// @formance/formance-sdk/sdk/models/shared/attempt.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Attempt$ = void 0;
const webhooksconfig_1 = require("./webhooksconfig");
const z = __importStar(require("zod"));
/** @internal */
var Attempt$;
(function (Attempt$) {
    Attempt$.inboundSchema = z.object({
        config: webhooksconfig_1.WebhooksConfig$.inboundSchema,
        createdAt: z
            .string()
            .datetime({ offset: true })
            .transform((v) => new Date(v)),
        id: z.string(),
        nextRetryAfter: z
            .string()
            .datetime({ offset: true })
            .transform((v) => new Date(v))
            .optional(),
        payload: z.string(),
        retryAttempt: z.number().int(),
        status: z.string(),
        statusCode: z.number().int(),
        updatedAt: z
            .string()
            .datetime({ offset: true })
            .transform((v) => new Date(v)),
        webhookID: z.string(),
    });
    Attempt$.outboundSchema = z.object({
        config: webhooksconfig_1.WebhooksConfig$.outboundSchema,
        createdAt: z.date().transform((v) => v.toISOString()),
        id: z.string(),
        nextRetryAfter: z
            .date()
            .transform((v) => v.toISOString())
            .optional(),
        payload: z.string(),
        retryAttempt: z.number().int(),
        status: z.string(),
        statusCode: z.number().int(),
        updatedAt: z.date().transform((v) => v.toISOString()),
        webhookID: z.string(),
    });
})(Attempt$ || (exports.Attempt$ = Attempt$ = {}));
