// @formance/formance-sdk/sdk/models/shared/balancescursor.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BalancesCursor$ = exports.BalancesCursorCursor$ = void 0;
const accountbalance_1 = require("./accountbalance");
const z = __importStar(require("zod"));
/** @internal */
var BalancesCursorCursor$;
(function (BalancesCursorCursor$) {
    BalancesCursorCursor$.inboundSchema = z.object({
        data: z.array(accountbalance_1.AccountBalance$.inboundSchema),
        hasMore: z.boolean(),
        next: z.string().optional(),
        pageSize: z.number().int(),
        previous: z.string().optional(),
    });
    BalancesCursorCursor$.outboundSchema = z.object({
        data: z.array(accountbalance_1.AccountBalance$.outboundSchema),
        hasMore: z.boolean(),
        next: z.string().optional(),
        pageSize: z.number().int(),
        previous: z.string().optional(),
    });
})(BalancesCursorCursor$ || (exports.BalancesCursorCursor$ = BalancesCursorCursor$ = {}));
/** @internal */
var BalancesCursor$;
(function (BalancesCursor$) {
    BalancesCursor$.inboundSchema = z.object({
        cursor: z.lazy(() => BalancesCursorCursor$.inboundSchema),
    });
    BalancesCursor$.outboundSchema = z.object({
        cursor: z.lazy(() => BalancesCursorCursor$.outboundSchema),
    });
})(BalancesCursor$ || (exports.BalancesCursor$ = BalancesCursor$ = {}));
