// @formance/formance-sdk/sdk/models/shared/bankaccount.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BankAccount$ = void 0;
const bankaccountrelatedaccounts_1 = require("./bankaccountrelatedaccounts");
const z = __importStar(require("zod"));
/** @internal */
var BankAccount$;
(function (BankAccount$) {
    BankAccount$.inboundSchema = z.object({
        accountID: z.string().optional(),
        accountNumber: z.string().optional(),
        connectorID: z.string(),
        country: z.string(),
        createdAt: z
            .string()
            .datetime({ offset: true })
            .transform((v) => new Date(v)),
        iban: z.string().optional(),
        id: z.string(),
        metadata: z.nullable(z.record(z.string())).optional(),
        name: z.string(),
        provider: z.string().optional(),
        relatedAccounts: z.array(bankaccountrelatedaccounts_1.BankAccountRelatedAccounts$.inboundSchema).optional(),
        swiftBicCode: z.string().optional(),
    });
    BankAccount$.outboundSchema = z.object({
        accountID: z.string().optional(),
        accountNumber: z.string().optional(),
        connectorID: z.string(),
        country: z.string(),
        createdAt: z.date().transform((v) => v.toISOString()),
        iban: z.string().optional(),
        id: z.string(),
        metadata: z.nullable(z.record(z.string())).optional(),
        name: z.string(),
        provider: z.string().optional(),
        relatedAccounts: z.array(bankaccountrelatedaccounts_1.BankAccountRelatedAccounts$.outboundSchema).optional(),
        swiftBicCode: z.string().optional(),
    });
})(BankAccount$ || (exports.BankAccount$ = BankAccount$ = {}));
