// @formance/formance-sdk/sdk/models/shared/bankaccountscursor.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BankAccountsCursor$ = exports.BankAccountsCursorCursor$ = void 0;
const bankaccount_1 = require("./bankaccount");
const z = __importStar(require("zod"));
/** @internal */
var BankAccountsCursorCursor$;
(function (BankAccountsCursorCursor$) {
    BankAccountsCursorCursor$.inboundSchema = z.object({
        data: z.array(bankaccount_1.BankAccount$.inboundSchema),
        hasMore: z.boolean(),
        next: z.string().optional(),
        pageSize: z.number().int(),
        previous: z.string().optional(),
    });
    BankAccountsCursorCursor$.outboundSchema = z.object({
        data: z.array(bankaccount_1.BankAccount$.outboundSchema),
        hasMore: z.boolean(),
        next: z.string().optional(),
        pageSize: z.number().int(),
        previous: z.string().optional(),
    });
})(BankAccountsCursorCursor$ || (exports.BankAccountsCursorCursor$ = BankAccountsCursorCursor$ = {}));
/** @internal */
var BankAccountsCursor$;
(function (BankAccountsCursor$) {
    BankAccountsCursor$.inboundSchema = z.object({
        cursor: z.lazy(() => BankAccountsCursorCursor$.inboundSchema),
    });
    BankAccountsCursor$.outboundSchema = z.object({
        cursor: z.lazy(() => BankAccountsCursorCursor$.outboundSchema),
    });
})(BankAccountsCursor$ || (exports.BankAccountsCursor$ = BankAccountsCursor$ = {}));
