// @formance/formance-sdk/sdk/models/shared/connectorsconfigsresponse.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConnectorsConfigsResponse$ = exports.ConnectorsConfigsResponseData$ = exports.ConnectorsConfigsResponseConnector$ = exports.Key$ = void 0;
const z = __importStar(require("zod"));
/** @internal */
var Key$;
(function (Key$) {
    Key$.inboundSchema = z.object({
        dataType: z.string(),
        required: z.boolean(),
    });
    Key$.outboundSchema = z.object({
        dataType: z.string(),
        required: z.boolean(),
    });
})(Key$ || (exports.Key$ = Key$ = {}));
/** @internal */
var ConnectorsConfigsResponseConnector$;
(function (ConnectorsConfigsResponseConnector$) {
    ConnectorsConfigsResponseConnector$.inboundSchema = z.object({
        key: z.lazy(() => Key$.inboundSchema),
    });
    ConnectorsConfigsResponseConnector$.outboundSchema = z.object({
        key: z.lazy(() => Key$.outboundSchema),
    });
})(ConnectorsConfigsResponseConnector$ || (exports.ConnectorsConfigsResponseConnector$ = ConnectorsConfigsResponseConnector$ = {}));
/** @internal */
var ConnectorsConfigsResponseData$;
(function (ConnectorsConfigsResponseData$) {
    ConnectorsConfigsResponseData$.inboundSchema = z.object({
        connector: z.lazy(() => ConnectorsConfigsResponseConnector$.inboundSchema),
    });
    ConnectorsConfigsResponseData$.outboundSchema = z.object({
        connector: z.lazy(() => ConnectorsConfigsResponseConnector$.outboundSchema),
    });
})(ConnectorsConfigsResponseData$ || (exports.ConnectorsConfigsResponseData$ = ConnectorsConfigsResponseData$ = {}));
/** @internal */
var ConnectorsConfigsResponse$;
(function (ConnectorsConfigsResponse$) {
    ConnectorsConfigsResponse$.inboundSchema = z.object({
        data: z.lazy(() => ConnectorsConfigsResponseData$.inboundSchema),
    });
    ConnectorsConfigsResponse$.outboundSchema = z.object({
        data: z.lazy(() => ConnectorsConfigsResponseData$.outboundSchema),
    });
})(ConnectorsConfigsResponse$ || (exports.ConnectorsConfigsResponse$ = ConnectorsConfigsResponse$ = {}));
