// @formance/formance-sdk/sdk/models/shared/getwalletsummaryresponse.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetWalletSummaryResponse$ = void 0;
const balancewithassets_1 = require("./balancewithassets");
const z = __importStar(require("zod"));
/** @internal */
var GetWalletSummaryResponse$;
(function (GetWalletSummaryResponse$) {
    GetWalletSummaryResponse$.inboundSchema = z.object({
        availableFunds: z.record(z.number().transform((v) => BigInt(v))),
        balances: z.array(balancewithassets_1.BalanceWithAssets$.inboundSchema),
        expirableFunds: z.record(z.number().transform((v) => BigInt(v))),
        expiredFunds: z.record(z.number().transform((v) => BigInt(v))),
        holdFunds: z.record(z.number().transform((v) => BigInt(v))),
    });
    GetWalletSummaryResponse$.outboundSchema = z.object({
        availableFunds: z.record(z.bigint().transform((v) => Number(v))),
        balances: z.array(balancewithassets_1.BalanceWithAssets$.outboundSchema),
        expirableFunds: z.record(z.bigint().transform((v) => Number(v))),
        expiredFunds: z.record(z.bigint().transform((v) => Number(v))),
        holdFunds: z.record(z.bigint().transform((v) => Number(v))),
    });
})(GetWalletSummaryResponse$ || (exports.GetWalletSummaryResponse$ = GetWalletSummaryResponse$ = {}));
