// @formance/formance-sdk/sdk/models/shared/log.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Log$ = exports.Type$ = exports.Type = void 0;
const z = __importStar(require("zod"));
var Type;
(function (Type) {
    Type["NewTransaction"] = "NEW_TRANSACTION";
    Type["SetMetadata"] = "SET_METADATA";
})(Type || (exports.Type = Type = {}));
/** @internal */
var Type$;
(function (Type$) {
    Type$.inboundSchema = z.nativeEnum(Type);
    Type$.outboundSchema = Type$.inboundSchema;
})(Type$ || (exports.Type$ = Type$ = {}));
/** @internal */
var Log$;
(function (Log$) {
    Log$.inboundSchema = z.object({
        data: z.record(z.any()),
        date: z
            .string()
            .datetime({ offset: true })
            .transform((v) => new Date(v)),
        hash: z.string(),
        id: z.number().int(),
        type: Type$.inboundSchema,
    });
    Log$.outboundSchema = z.object({
        data: z.record(z.any()),
        date: z.date().transform((v) => v.toISOString()),
        hash: z.string(),
        id: z.number().int(),
        type: Type$.outboundSchema,
    });
})(Log$ || (exports.Log$ = Log$ = {}));
