// @formance/formance-sdk/sdk/models/shared/orchestrationpaymentadjustment.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrchestrationPaymentAdjustment$ = exports.OrchestrationPaymentAdjustmentRaw$ = void 0;
const orchestrationpaymentstatus_1 = require("./orchestrationpaymentstatus");
const z = __importStar(require("zod"));
/** @internal */
var OrchestrationPaymentAdjustmentRaw$;
(function (OrchestrationPaymentAdjustmentRaw$) {
    OrchestrationPaymentAdjustmentRaw$.inboundSchema = z.object({});
    OrchestrationPaymentAdjustmentRaw$.outboundSchema = z.object({});
})(OrchestrationPaymentAdjustmentRaw$ || (exports.OrchestrationPaymentAdjustmentRaw$ = OrchestrationPaymentAdjustmentRaw$ = {}));
/** @internal */
var OrchestrationPaymentAdjustment$;
(function (OrchestrationPaymentAdjustment$) {
    OrchestrationPaymentAdjustment$.inboundSchema = z.object({
        absolute: z.boolean(),
        amount: z.number().transform((v) => BigInt(v)),
        date: z
            .string()
            .datetime({ offset: true })
            .transform((v) => new Date(v)),
        raw: z.lazy(() => OrchestrationPaymentAdjustmentRaw$.inboundSchema),
        status: orchestrationpaymentstatus_1.OrchestrationPaymentStatus$.inboundSchema,
    });
    OrchestrationPaymentAdjustment$.outboundSchema = z.object({
        absolute: z.boolean(),
        amount: z.bigint().transform((v) => Number(v)),
        date: z.date().transform((v) => v.toISOString()),
        raw: z.lazy(() => OrchestrationPaymentAdjustmentRaw$.outboundSchema),
        status: orchestrationpaymentstatus_1.OrchestrationPaymentStatus$.outboundSchema,
    });
})(OrchestrationPaymentAdjustment$ || (exports.OrchestrationPaymentAdjustment$ = OrchestrationPaymentAdjustment$ = {}));
