// @formance/formance-sdk/sdk/models/shared/payment.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Payment$ = exports.Raw$ = void 0;
const connector_1 = require("./connector");
const paymentadjustment_1 = require("./paymentadjustment");
const paymentscheme_1 = require("./paymentscheme");
const paymentstatus_1 = require("./paymentstatus");
const paymenttype_1 = require("./paymenttype");
const z = __importStar(require("zod"));
/** @internal */
var Raw$;
(function (Raw$) {
    Raw$.inboundSchema = z.object({});
    Raw$.outboundSchema = z.object({});
})(Raw$ || (exports.Raw$ = Raw$ = {}));
/** @internal */
var Payment$;
(function (Payment$) {
    Payment$.inboundSchema = z.object({
        adjustments: z.array(paymentadjustment_1.PaymentAdjustment$.inboundSchema),
        amount: z.number().transform((v) => BigInt(v)),
        asset: z.string(),
        connectorID: z.string(),
        createdAt: z
            .string()
            .datetime({ offset: true })
            .transform((v) => new Date(v)),
        destinationAccountID: z.string(),
        id: z.string(),
        initialAmount: z.number().transform((v) => BigInt(v)),
        metadata: z.nullable(z.record(z.string())),
        provider: connector_1.Connector$.inboundSchema.optional(),
        raw: z.nullable(z.lazy(() => Raw$.inboundSchema)),
        reference: z.string(),
        scheme: paymentscheme_1.PaymentScheme$.inboundSchema,
        sourceAccountID: z.string(),
        status: paymentstatus_1.PaymentStatus$.inboundSchema,
        type: paymenttype_1.PaymentType$.inboundSchema,
    });
    Payment$.outboundSchema = z.object({
        adjustments: z.array(paymentadjustment_1.PaymentAdjustment$.outboundSchema),
        amount: z.bigint().transform((v) => Number(v)),
        asset: z.string(),
        connectorID: z.string(),
        createdAt: z.date().transform((v) => v.toISOString()),
        destinationAccountID: z.string(),
        id: z.string(),
        initialAmount: z.bigint().transform((v) => Number(v)),
        metadata: z.nullable(z.record(z.string())),
        provider: connector_1.Connector$.outboundSchema.optional(),
        raw: z.nullable(z.lazy(() => Raw$.outboundSchema)),
        reference: z.string(),
        scheme: paymentscheme_1.PaymentScheme$.outboundSchema,
        sourceAccountID: z.string(),
        status: paymentstatus_1.PaymentStatus$.outboundSchema,
        type: paymenttype_1.PaymentType$.outboundSchema,
    });
})(Payment$ || (exports.Payment$ = Payment$ = {}));
