// @formance/formance-sdk/sdk/models/shared/paymentscheme.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentScheme$ = exports.PaymentScheme = void 0;
const z = __importStar(require("zod"));
var PaymentScheme;
(function (PaymentScheme) {
    PaymentScheme["Unknown"] = "unknown";
    PaymentScheme["Other"] = "other";
    PaymentScheme["Visa"] = "visa";
    PaymentScheme["Mastercard"] = "mastercard";
    PaymentScheme["Amex"] = "amex";
    PaymentScheme["Diners"] = "diners";
    PaymentScheme["Discover"] = "discover";
    PaymentScheme["Jcb"] = "jcb";
    PaymentScheme["Unionpay"] = "unionpay";
    PaymentScheme["Alipay"] = "alipay";
    PaymentScheme["Cup"] = "cup";
    PaymentScheme["SepaDebit"] = "sepa debit";
    PaymentScheme["SepaCredit"] = "sepa credit";
    PaymentScheme["Sepa"] = "sepa";
    PaymentScheme["ApplePay"] = "apple pay";
    PaymentScheme["GooglePay"] = "google pay";
    PaymentScheme["Doku"] = "doku";
    PaymentScheme["Dragonpay"] = "dragonpay";
    PaymentScheme["Maestro"] = "maestro";
    PaymentScheme["Molpay"] = "molpay";
    PaymentScheme["A2a"] = "a2a";
    PaymentScheme["AchDebit"] = "ach debit";
    PaymentScheme["Ach"] = "ach";
    PaymentScheme["Rtp"] = "rtp";
})(PaymentScheme || (exports.PaymentScheme = PaymentScheme = {}));
/** @internal */
var PaymentScheme$;
(function (PaymentScheme$) {
    PaymentScheme$.inboundSchema = z.nativeEnum(PaymentScheme);
    PaymentScheme$.outboundSchema = PaymentScheme$.inboundSchema;
})(PaymentScheme$ || (exports.PaymentScheme$ = PaymentScheme$ = {}));
