// @formance/formance-sdk/sdk/models/shared/paymentadjustment.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentAdjustment$ = exports.PaymentAdjustmentRaw$ = void 0;
const paymentstatus_1 = require("./paymentstatus");
const z = __importStar(require("zod"));
/** @internal */
var PaymentAdjustmentRaw$;
(function (PaymentAdjustmentRaw$) {
    PaymentAdjustmentRaw$.inboundSchema = z.object({});
    PaymentAdjustmentRaw$.outboundSchema = z.object({});
})(PaymentAdjustmentRaw$ || (exports.PaymentAdjustmentRaw$ = PaymentAdjustmentRaw$ = {}));
/** @internal */
var PaymentAdjustment$;
(function (PaymentAdjustment$) {
    PaymentAdjustment$.inboundSchema = z.object({
        amount: z.number().transform((v) => BigInt(v)),
        createdAt: z
            .string()
            .datetime({ offset: true })
            .transform((v) => new Date(v)),
        raw: z.lazy(() => PaymentAdjustmentRaw$.inboundSchema),
        reference: z.string(),
        status: paymentstatus_1.PaymentStatus$.inboundSchema,
    });
    PaymentAdjustment$.outboundSchema = z.object({
        amount: z.bigint().transform((v) => Number(v)),
        createdAt: z.date().transform((v) => v.toISOString()),
        raw: z.lazy(() => PaymentAdjustmentRaw$.outboundSchema),
        reference: z.string(),
        status: paymentstatus_1.PaymentStatus$.outboundSchema,
    });
})(PaymentAdjustment$ || (exports.PaymentAdjustment$ = PaymentAdjustment$ = {}));
