// @formance/formance-sdk/sdk/models/shared/paymentsaccount.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentsAccount$ = exports.PaymentsAccountRaw$ = void 0;
const accounttype_1 = require("./accounttype");
const z = __importStar(require("zod"));
/** @internal */
var PaymentsAccountRaw$;
(function (PaymentsAccountRaw$) {
    PaymentsAccountRaw$.inboundSchema = z.object({});
    PaymentsAccountRaw$.outboundSchema = z.object({});
})(PaymentsAccountRaw$ || (exports.PaymentsAccountRaw$ = PaymentsAccountRaw$ = {}));
/** @internal */
var PaymentsAccount$;
(function (PaymentsAccount$) {
    PaymentsAccount$.inboundSchema = z.object({
        accountName: z.string(),
        connectorID: z.string(),
        createdAt: z
            .string()
            .datetime({ offset: true })
            .transform((v) => new Date(v)),
        defaultAsset: z.string(),
        defaultCurrency: z.string(),
        id: z.string(),
        metadata: z.nullable(z.record(z.string())),
        pools: z.array(z.string()).optional(),
        provider: z.string().optional(),
        raw: z.nullable(z.lazy(() => PaymentsAccountRaw$.inboundSchema)),
        reference: z.string(),
        type: accounttype_1.AccountType$.inboundSchema,
    });
    PaymentsAccount$.outboundSchema = z.object({
        accountName: z.string(),
        connectorID: z.string(),
        createdAt: z.date().transform((v) => v.toISOString()),
        defaultAsset: z.string(),
        defaultCurrency: z.string(),
        id: z.string(),
        metadata: z.nullable(z.record(z.string())),
        pools: z.array(z.string()).optional(),
        provider: z.string().optional(),
        raw: z.nullable(z.lazy(() => PaymentsAccountRaw$.outboundSchema)),
        reference: z.string(),
        type: accounttype_1.AccountType$.outboundSchema,
    });
})(PaymentsAccount$ || (exports.PaymentsAccount$ = PaymentsAccount$ = {}));
