// @formance/formance-sdk/sdk/models/shared/reconciliation.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Reconciliation$ = void 0;
const z = __importStar(require("zod"));
/** @internal */
var Reconciliation$;
(function (Reconciliation$) {
    Reconciliation$.inboundSchema = z.object({
        createdAt: z
            .string()
            .datetime({ offset: true })
            .transform((v) => new Date(v)),
        driftBalances: z.record(z.number().transform((v) => BigInt(v))),
        error: z.string().optional(),
        id: z.string(),
        ledgerBalances: z.record(z.number().transform((v) => BigInt(v))),
        paymentsBalances: z.record(z.number().transform((v) => BigInt(v))),
        policyID: z.string(),
        reconciledAtLedger: z
            .string()
            .datetime({ offset: true })
            .transform((v) => new Date(v)),
        reconciledAtPayments: z
            .string()
            .datetime({ offset: true })
            .transform((v) => new Date(v)),
        status: z.string(),
    });
    Reconciliation$.outboundSchema = z.object({
        createdAt: z.date().transform((v) => v.toISOString()),
        driftBalances: z.record(z.bigint().transform((v) => Number(v))),
        error: z.string().optional(),
        id: z.string(),
        ledgerBalances: z.record(z.bigint().transform((v) => Number(v))),
        paymentsBalances: z.record(z.bigint().transform((v) => Number(v))),
        policyID: z.string(),
        reconciledAtLedger: z.date().transform((v) => v.toISOString()),
        reconciledAtPayments: z.date().transform((v) => v.toISOString()),
        status: z.string(),
    });
})(Reconciliation$ || (exports.Reconciliation$ = Reconciliation$ = {}));
