// @formance/formance-sdk/sdk/models/shared/response.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Response$ = exports.ResponseCursor$ = exports.Total$ = void 0;
const z = __importStar(require("zod"));
/** @internal */
var Total$;
(function (Total$) {
    Total$.inboundSchema = z.object({
        relation: z.string().optional(),
        value: z.number().int().optional(),
    });
    Total$.outboundSchema = z.object({
        relation: z.string().optional(),
        value: z.number().int().optional(),
    });
})(Total$ || (exports.Total$ = Total$ = {}));
/** @internal */
var ResponseCursor$;
(function (ResponseCursor$) {
    ResponseCursor$.inboundSchema = z.object({
        data: z.array(z.record(z.any())).optional(),
        hasMore: z.boolean().optional(),
        next: z.string().optional(),
        pageSize: z.number().int().optional(),
        previous: z.string().optional(),
        total: z.lazy(() => Total$.inboundSchema).optional(),
    });
    ResponseCursor$.outboundSchema = z.object({
        data: z.array(z.record(z.any())).optional(),
        hasMore: z.boolean().optional(),
        next: z.string().optional(),
        pageSize: z.number().int().optional(),
        previous: z.string().optional(),
        total: z.lazy(() => Total$.outboundSchema).optional(),
    });
})(ResponseCursor$ || (exports.ResponseCursor$ = ResponseCursor$ = {}));
/** @internal */
var Response$;
(function (Response$) {
    Response$.inboundSchema = z.object({
        cursor: z.lazy(() => ResponseCursor$.inboundSchema).optional(),
        data: z.record(z.any()).optional(),
    });
    Response$.outboundSchema = z.object({
        cursor: z.lazy(() => ResponseCursor$.outboundSchema).optional(),
        data: z.record(z.any()).optional(),
    });
})(Response$ || (exports.Response$ = Response$ = {}));
