// @formance/formance-sdk/sdk/models/shared/stagesend.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.StageSend$ = void 0;
const monetary_1 = require("./monetary");
const stagesenddestination_1 = require("./stagesenddestination");
const stagesendsource_1 = require("./stagesendsource");
const z = __importStar(require("zod"));
/** @internal */
var StageSend$;
(function (StageSend$) {
    StageSend$.inboundSchema = z.object({
        amount: monetary_1.Monetary$.inboundSchema.optional(),
        destination: stagesenddestination_1.StageSendDestination$.inboundSchema.optional(),
        metadata: z.record(z.string()).optional(),
        source: stagesendsource_1.StageSendSource$.inboundSchema.optional(),
        timestamp: z
            .string()
            .datetime({ offset: true })
            .transform((v) => new Date(v))
            .optional(),
    });
    StageSend$.outboundSchema = z.object({
        amount: monetary_1.Monetary$.outboundSchema.optional(),
        destination: stagesenddestination_1.StageSendDestination$.outboundSchema.optional(),
        metadata: z.record(z.string()).optional(),
        source: stagesendsource_1.StageSendSource$.outboundSchema.optional(),
        timestamp: z
            .date()
            .transform((v) => v.toISOString())
            .optional(),
    });
})(StageSend$ || (exports.StageSend$ = StageSend$ = {}));
