// @formance/formance-sdk/sdk/models/shared/taskscursor.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TasksCursor$ = exports.TasksCursorCursor$ = exports.TasksCursorData$ = void 0;
const taskbankingcircle_1 = require("./taskbankingcircle");
const taskcurrencycloud_1 = require("./taskcurrencycloud");
const taskdummypay_1 = require("./taskdummypay");
const taskmangopay_1 = require("./taskmangopay");
const taskmodulr_1 = require("./taskmodulr");
const taskmoneycorp_1 = require("./taskmoneycorp");
const taskstripe_1 = require("./taskstripe");
const taskwise_1 = require("./taskwise");
const z = __importStar(require("zod"));
/** @internal */
var TasksCursorData$;
(function (TasksCursorData$) {
    TasksCursorData$.inboundSchema = z.union([
        taskstripe_1.TaskStripe$.inboundSchema,
        taskwise_1.TaskWise$.inboundSchema,
        taskcurrencycloud_1.TaskCurrencyCloud$.inboundSchema,
        taskdummypay_1.TaskDummyPay$.inboundSchema,
        taskmodulr_1.TaskModulr$.inboundSchema,
        taskbankingcircle_1.TaskBankingCircle$.inboundSchema,
        taskmangopay_1.TaskMangoPay$.inboundSchema,
        taskmoneycorp_1.TaskMoneycorp$.inboundSchema,
    ]);
    TasksCursorData$.outboundSchema = z.union([
        taskstripe_1.TaskStripe$.outboundSchema,
        taskwise_1.TaskWise$.outboundSchema,
        taskcurrencycloud_1.TaskCurrencyCloud$.outboundSchema,
        taskdummypay_1.TaskDummyPay$.outboundSchema,
        taskmodulr_1.TaskModulr$.outboundSchema,
        taskbankingcircle_1.TaskBankingCircle$.outboundSchema,
        taskmangopay_1.TaskMangoPay$.outboundSchema,
        taskmoneycorp_1.TaskMoneycorp$.outboundSchema,
    ]);
})(TasksCursorData$ || (exports.TasksCursorData$ = TasksCursorData$ = {}));
/** @internal */
var TasksCursorCursor$;
(function (TasksCursorCursor$) {
    TasksCursorCursor$.inboundSchema = z.object({
        data: z.array(z.union([
            taskstripe_1.TaskStripe$.inboundSchema,
            taskwise_1.TaskWise$.inboundSchema,
            taskcurrencycloud_1.TaskCurrencyCloud$.inboundSchema,
            taskdummypay_1.TaskDummyPay$.inboundSchema,
            taskmodulr_1.TaskModulr$.inboundSchema,
            taskbankingcircle_1.TaskBankingCircle$.inboundSchema,
            taskmangopay_1.TaskMangoPay$.inboundSchema,
            taskmoneycorp_1.TaskMoneycorp$.inboundSchema,
        ])),
        hasMore: z.boolean(),
        next: z.string().optional(),
        pageSize: z.number().int(),
        previous: z.string().optional(),
    });
    TasksCursorCursor$.outboundSchema = z.object({
        data: z.array(z.union([
            taskstripe_1.TaskStripe$.outboundSchema,
            taskwise_1.TaskWise$.outboundSchema,
            taskcurrencycloud_1.TaskCurrencyCloud$.outboundSchema,
            taskdummypay_1.TaskDummyPay$.outboundSchema,
            taskmodulr_1.TaskModulr$.outboundSchema,
            taskbankingcircle_1.TaskBankingCircle$.outboundSchema,
            taskmangopay_1.TaskMangoPay$.outboundSchema,
            taskmoneycorp_1.TaskMoneycorp$.outboundSchema,
        ])),
        hasMore: z.boolean(),
        next: z.string().optional(),
        pageSize: z.number().int(),
        previous: z.string().optional(),
    });
})(TasksCursorCursor$ || (exports.TasksCursorCursor$ = TasksCursorCursor$ = {}));
/** @internal */
var TasksCursor$;
(function (TasksCursor$) {
    TasksCursor$.inboundSchema = z.object({
        cursor: z.lazy(() => TasksCursorCursor$.inboundSchema),
    });
    TasksCursor$.outboundSchema = z.object({
        cursor: z.lazy(() => TasksCursorCursor$.outboundSchema),
    });
})(TasksCursor$ || (exports.TasksCursor$ = TasksCursor$ = {}));
