// @formance/formance-sdk/sdk/models/shared/taskcurrencycloud.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TaskCurrencyCloud$ = exports.TaskCurrencyCloudState$ = exports.TaskCurrencyCloudDescriptor$ = void 0;
const paymentstatus_1 = require("./paymentstatus");
const z = __importStar(require("zod"));
/** @internal */
var TaskCurrencyCloudDescriptor$;
(function (TaskCurrencyCloudDescriptor$) {
    TaskCurrencyCloudDescriptor$.inboundSchema = z.object({
        name: z.string().optional(),
    });
    TaskCurrencyCloudDescriptor$.outboundSchema = z.object({
        name: z.string().optional(),
    });
})(TaskCurrencyCloudDescriptor$ || (exports.TaskCurrencyCloudDescriptor$ = TaskCurrencyCloudDescriptor$ = {}));
/** @internal */
var TaskCurrencyCloudState$;
(function (TaskCurrencyCloudState$) {
    TaskCurrencyCloudState$.inboundSchema = z.object({});
    TaskCurrencyCloudState$.outboundSchema = z.object({});
})(TaskCurrencyCloudState$ || (exports.TaskCurrencyCloudState$ = TaskCurrencyCloudState$ = {}));
/** @internal */
var TaskCurrencyCloud$;
(function (TaskCurrencyCloud$) {
    TaskCurrencyCloud$.inboundSchema = z.object({
        connectorID: z.string(),
        createdAt: z
            .string()
            .datetime({ offset: true })
            .transform((v) => new Date(v)),
        descriptor: z.lazy(() => TaskCurrencyCloudDescriptor$.inboundSchema),
        error: z.string().optional(),
        id: z.string(),
        state: z.lazy(() => TaskCurrencyCloudState$.inboundSchema),
        status: paymentstatus_1.PaymentStatus$.inboundSchema,
        updatedAt: z
            .string()
            .datetime({ offset: true })
            .transform((v) => new Date(v)),
    });
    TaskCurrencyCloud$.outboundSchema = z.object({
        connectorID: z.string(),
        createdAt: z.date().transform((v) => v.toISOString()),
        descriptor: z.lazy(() => TaskCurrencyCloudDescriptor$.outboundSchema),
        error: z.string().optional(),
        id: z.string(),
        state: z.lazy(() => TaskCurrencyCloudState$.outboundSchema),
        status: paymentstatus_1.PaymentStatus$.outboundSchema,
        updatedAt: z.date().transform((v) => v.toISOString()),
    });
})(TaskCurrencyCloud$ || (exports.TaskCurrencyCloud$ = TaskCurrencyCloud$ = {}));
