// @formance/formance-sdk/sdk/models/shared/taskmangopay.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TaskMangoPay$ = exports.TaskMangoPayState$ = exports.TaskMangoPayDescriptor$ = void 0;
const paymentstatus_1 = require("./paymentstatus");
const z = __importStar(require("zod"));
/** @internal */
var TaskMangoPayDescriptor$;
(function (TaskMangoPayDescriptor$) {
    TaskMangoPayDescriptor$.inboundSchema = z.object({
        key: z.string().optional(),
        name: z.string().optional(),
        userID: z.string().optional(),
    });
    TaskMangoPayDescriptor$.outboundSchema = z.object({
        key: z.string().optional(),
        name: z.string().optional(),
        userID: z.string().optional(),
    });
})(TaskMangoPayDescriptor$ || (exports.TaskMangoPayDescriptor$ = TaskMangoPayDescriptor$ = {}));
/** @internal */
var TaskMangoPayState$;
(function (TaskMangoPayState$) {
    TaskMangoPayState$.inboundSchema = z.object({});
    TaskMangoPayState$.outboundSchema = z.object({});
})(TaskMangoPayState$ || (exports.TaskMangoPayState$ = TaskMangoPayState$ = {}));
/** @internal */
var TaskMangoPay$;
(function (TaskMangoPay$) {
    TaskMangoPay$.inboundSchema = z.object({
        connectorID: z.string(),
        createdAt: z
            .string()
            .datetime({ offset: true })
            .transform((v) => new Date(v)),
        descriptor: z.lazy(() => TaskMangoPayDescriptor$.inboundSchema),
        error: z.string().optional(),
        id: z.string(),
        state: z.lazy(() => TaskMangoPayState$.inboundSchema),
        status: paymentstatus_1.PaymentStatus$.inboundSchema,
        updatedAt: z
            .string()
            .datetime({ offset: true })
            .transform((v) => new Date(v)),
    });
    TaskMangoPay$.outboundSchema = z.object({
        connectorID: z.string(),
        createdAt: z.date().transform((v) => v.toISOString()),
        descriptor: z.lazy(() => TaskMangoPayDescriptor$.outboundSchema),
        error: z.string().optional(),
        id: z.string(),
        state: z.lazy(() => TaskMangoPayState$.outboundSchema),
        status: paymentstatus_1.PaymentStatus$.outboundSchema,
        updatedAt: z.date().transform((v) => v.toISOString()),
    });
})(TaskMangoPay$ || (exports.TaskMangoPay$ = TaskMangoPay$ = {}));
