// @formance/formance-sdk/sdk/models/shared/taskbankingcircle.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TaskBankingCircle$ = exports.TaskBankingCircleState$ = exports.Descriptor$ = void 0;
const paymentstatus_1 = require("./paymentstatus");
const z = __importStar(require("zod"));
/** @internal */
var Descriptor$;
(function (Descriptor$) {
    Descriptor$.inboundSchema = z.object({
        key: z.string().optional(),
        name: z.string().optional(),
    });
    Descriptor$.outboundSchema = z.object({
        key: z.string().optional(),
        name: z.string().optional(),
    });
})(Descriptor$ || (exports.Descriptor$ = Descriptor$ = {}));
/** @internal */
var TaskBankingCircleState$;
(function (TaskBankingCircleState$) {
    TaskBankingCircleState$.inboundSchema = z.object({});
    TaskBankingCircleState$.outboundSchema = z.object({});
})(TaskBankingCircleState$ || (exports.TaskBankingCircleState$ = TaskBankingCircleState$ = {}));
/** @internal */
var TaskBankingCircle$;
(function (TaskBankingCircle$) {
    TaskBankingCircle$.inboundSchema = z.object({
        connectorID: z.string(),
        createdAt: z
            .string()
            .datetime({ offset: true })
            .transform((v) => new Date(v)),
        descriptor: z.lazy(() => Descriptor$.inboundSchema),
        error: z.string().optional(),
        id: z.string(),
        state: z.lazy(() => TaskBankingCircleState$.inboundSchema),
        status: paymentstatus_1.PaymentStatus$.inboundSchema,
        updatedAt: z
            .string()
            .datetime({ offset: true })
            .transform((v) => new Date(v)),
    });
    TaskBankingCircle$.outboundSchema = z.object({
        connectorID: z.string(),
        createdAt: z.date().transform((v) => v.toISOString()),
        descriptor: z.lazy(() => Descriptor$.outboundSchema),
        error: z.string().optional(),
        id: z.string(),
        state: z.lazy(() => TaskBankingCircleState$.outboundSchema),
        status: paymentstatus_1.PaymentStatus$.outboundSchema,
        updatedAt: z.date().transform((v) => v.toISOString()),
    });
})(TaskBankingCircle$ || (exports.TaskBankingCircle$ = TaskBankingCircle$ = {}));
