// @formance/formance-sdk/sdk/models/shared/transferinitiationrequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransferInitiationRequest$ = exports.TransferInitiationRequestType$ = exports.TransferInitiationRequestType = void 0;
const connector_1 = require("./connector");
const z = __importStar(require("zod"));
var TransferInitiationRequestType;
(function (TransferInitiationRequestType) {
    TransferInitiationRequestType["Transfer"] = "TRANSFER";
    TransferInitiationRequestType["Payout"] = "PAYOUT";
})(TransferInitiationRequestType || (exports.TransferInitiationRequestType = TransferInitiationRequestType = {}));
/** @internal */
var TransferInitiationRequestType$;
(function (TransferInitiationRequestType$) {
    TransferInitiationRequestType$.inboundSchema = z.nativeEnum(TransferInitiationRequestType);
    TransferInitiationRequestType$.outboundSchema = TransferInitiationRequestType$.inboundSchema;
})(TransferInitiationRequestType$ || (exports.TransferInitiationRequestType$ = TransferInitiationRequestType$ = {}));
/** @internal */
var TransferInitiationRequest$;
(function (TransferInitiationRequest$) {
    TransferInitiationRequest$.inboundSchema = z.object({
        amount: z.number().transform((v) => BigInt(v)),
        asset: z.string(),
        connectorID: z.string().optional(),
        description: z.string(),
        destinationAccountID: z.string(),
        metadata: z.nullable(z.record(z.string())).optional(),
        provider: connector_1.Connector$.inboundSchema.optional(),
        reference: z.string(),
        scheduledAt: z
            .string()
            .datetime({ offset: true })
            .transform((v) => new Date(v)),
        sourceAccountID: z.string(),
        type: TransferInitiationRequestType$.inboundSchema,
        validated: z.boolean(),
    });
    TransferInitiationRequest$.outboundSchema = z.object({
        amount: z.bigint().transform((v) => Number(v)),
        asset: z.string(),
        connectorID: z.string().optional(),
        description: z.string(),
        destinationAccountID: z.string(),
        metadata: z.nullable(z.record(z.string())).optional(),
        provider: connector_1.Connector$.outboundSchema.optional(),
        reference: z.string(),
        scheduledAt: z.date().transform((v) => v.toISOString()),
        sourceAccountID: z.string(),
        type: TransferInitiationRequestType$.outboundSchema,
        validated: z.boolean(),
    });
})(TransferInitiationRequest$ || (exports.TransferInitiationRequest$ = TransferInitiationRequest$ = {}));
