// @formance/formance-sdk/sdk/models/shared/transferinitiationstatus.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransferInitiationStatus$ = exports.TransferInitiationStatus = void 0;
const z = __importStar(require("zod"));
var TransferInitiationStatus;
(function (TransferInitiationStatus) {
    TransferInitiationStatus["WaitingForValidation"] = "WAITING_FOR_VALIDATION";
    TransferInitiationStatus["Processing"] = "PROCESSING";
    TransferInitiationStatus["Processed"] = "PROCESSED";
    TransferInitiationStatus["Failed"] = "FAILED";
    TransferInitiationStatus["Rejected"] = "REJECTED";
    TransferInitiationStatus["Validated"] = "VALIDATED";
    TransferInitiationStatus["AskRetried"] = "ASK_RETRIED";
    TransferInitiationStatus["AskReversed"] = "ASK_REVERSED";
    TransferInitiationStatus["ReverseProcessing"] = "REVERSE_PROCESSING";
    TransferInitiationStatus["ReverseFailed"] = "REVERSE_FAILED";
    TransferInitiationStatus["PartiallyReversed"] = "PARTIALLY_REVERSED";
    TransferInitiationStatus["Reversed"] = "REVERSED";
})(TransferInitiationStatus || (exports.TransferInitiationStatus = TransferInitiationStatus = {}));
/** @internal */
var TransferInitiationStatus$;
(function (TransferInitiationStatus$) {
    TransferInitiationStatus$.inboundSchema = z.nativeEnum(TransferInitiationStatus);
    TransferInitiationStatus$.outboundSchema = TransferInitiationStatus$.inboundSchema;
})(TransferInitiationStatus$ || (exports.TransferInitiationStatus$ = TransferInitiationStatus$ = {}));
