// @formance/formance-sdk/sdk/models/shared/transferinitiationadjusments.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransferInitiationAdjusments$ = void 0;
const transferinitiationstatus_1 = require("./transferinitiationstatus");
const z = __importStar(require("zod"));
/** @internal */
var TransferInitiationAdjusments$;
(function (TransferInitiationAdjusments$) {
    TransferInitiationAdjusments$.inboundSchema = z.object({
        adjustmentID: z.string(),
        createdAt: z
            .string()
            .datetime({ offset: true })
            .transform((v) => new Date(v)),
        error: z.string(),
        metadata: z.nullable(z.record(z.string())).optional(),
        status: transferinitiationstatus_1.TransferInitiationStatus$.inboundSchema,
    });
    TransferInitiationAdjusments$.outboundSchema = z.object({
        adjustmentID: z.string(),
        createdAt: z.date().transform((v) => v.toISOString()),
        error: z.string(),
        metadata: z.nullable(z.record(z.string())).optional(),
        status: transferinitiationstatus_1.TransferInitiationStatus$.outboundSchema,
    });
})(TransferInitiationAdjusments$ || (exports.TransferInitiationAdjusments$ = TransferInitiationAdjusments$ = {}));
