// @formance/formance-sdk/sdk/models/shared/updatetransferinitiationstatusrequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateTransferInitiationStatusRequest$ = exports.Status$ = exports.Status = void 0;
const z = __importStar(require("zod"));
var Status;
(function (Status) {
    Status["WaitingForValidation"] = "WAITING_FOR_VALIDATION";
    Status["Processing"] = "PROCESSING";
    Status["Processed"] = "PROCESSED";
    Status["Failed"] = "FAILED";
    Status["Rejected"] = "REJECTED";
    Status["Validated"] = "VALIDATED";
})(Status || (exports.Status = Status = {}));
/** @internal */
var Status$;
(function (Status$) {
    Status$.inboundSchema = z.nativeEnum(Status);
    Status$.outboundSchema = Status$.inboundSchema;
})(Status$ || (exports.Status$ = Status$ = {}));
/** @internal */
var UpdateTransferInitiationStatusRequest$;
(function (UpdateTransferInitiationStatusRequest$) {
    UpdateTransferInitiationStatusRequest$.inboundSchema = z.object({
        status: Status$.inboundSchema,
    });
    UpdateTransferInitiationStatusRequest$.outboundSchema = z.object({
        status: Status$.outboundSchema,
    });
})(UpdateTransferInitiationStatusRequest$ || (exports.UpdateTransferInitiationStatusRequest$ = UpdateTransferInitiationStatusRequest$ = {}));
