// @formance/formance-sdk/sdk/models/shared/v2accountscursorresponse.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.V2AccountsCursorResponse$ = exports.V2AccountsCursorResponseCursor$ = void 0;
const v2account_1 = require("./v2account");
const z = __importStar(require("zod"));
/** @internal */
var V2AccountsCursorResponseCursor$;
(function (V2AccountsCursorResponseCursor$) {
    V2AccountsCursorResponseCursor$.inboundSchema = z.object({
        data: z.array(v2account_1.V2Account$.inboundSchema),
        hasMore: z.boolean(),
        next: z.string().optional(),
        pageSize: z.number().int(),
        previous: z.string().optional(),
    });
    V2AccountsCursorResponseCursor$.outboundSchema = z.object({
        data: z.array(v2account_1.V2Account$.outboundSchema),
        hasMore: z.boolean(),
        next: z.string().optional(),
        pageSize: z.number().int(),
        previous: z.string().optional(),
    });
})(V2AccountsCursorResponseCursor$ || (exports.V2AccountsCursorResponseCursor$ = V2AccountsCursorResponseCursor$ = {}));
/** @internal */
var V2AccountsCursorResponse$;
(function (V2AccountsCursorResponse$) {
    V2AccountsCursorResponse$.inboundSchema = z.object({
        cursor: z.lazy(() => V2AccountsCursorResponseCursor$.inboundSchema),
    });
    V2AccountsCursorResponse$.outboundSchema = z.object({
        cursor: z.lazy(() => V2AccountsCursorResponseCursor$.outboundSchema),
    });
})(V2AccountsCursorResponse$ || (exports.V2AccountsCursorResponse$ = V2AccountsCursorResponse$ = {}));
