// @formance/formance-sdk/sdk/models/shared/v2activitystripetransfer.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.V2ActivityStripeTransfer$ = exports.V2ActivityStripeTransferMetadata$ = void 0;
const z = __importStar(require("zod"));
/** @internal */
var V2ActivityStripeTransferMetadata$;
(function (V2ActivityStripeTransferMetadata$) {
    V2ActivityStripeTransferMetadata$.inboundSchema = z.object({});
    V2ActivityStripeTransferMetadata$.outboundSchema = z.object({});
})(V2ActivityStripeTransferMetadata$ || (exports.V2ActivityStripeTransferMetadata$ = V2ActivityStripeTransferMetadata$ = {}));
/** @internal */
var V2ActivityStripeTransfer$;
(function (V2ActivityStripeTransfer$) {
    V2ActivityStripeTransfer$.inboundSchema = z.object({
        amount: z
            .number()
            .transform((v) => BigInt(v))
            .optional(),
        asset: z.string().optional(),
        connectorID: z.string().optional(),
        destination: z.string().optional(),
        metadata: z.lazy(() => V2ActivityStripeTransferMetadata$.inboundSchema).optional(),
        waitingValidation: z.boolean().default(false),
    });
    V2ActivityStripeTransfer$.outboundSchema = z.object({
        amount: z
            .bigint()
            .transform((v) => Number(v))
            .optional(),
        asset: z.string().optional(),
        connectorID: z.string().optional(),
        destination: z.string().optional(),
        metadata: z.lazy(() => V2ActivityStripeTransferMetadata$.outboundSchema).optional(),
        waitingValidation: z.boolean().default(false),
    });
})(V2ActivityStripeTransfer$ || (exports.V2ActivityStripeTransfer$ = V2ActivityStripeTransfer$ = {}));
