// @formance/formance-sdk/sdk/models/shared/v2bulkelementaddmetadata.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.V2BulkElementAddMetadata$ = exports.V2BulkElementAddMetadataData$ = void 0;
const v2targetid_1 = require("./v2targetid");
const v2targettype_1 = require("./v2targettype");
const z = __importStar(require("zod"));
/** @internal */
var V2BulkElementAddMetadataData$;
(function (V2BulkElementAddMetadataData$) {
    V2BulkElementAddMetadataData$.inboundSchema = z.object({
        metadata: z.record(z.string()),
        targetId: v2targetid_1.V2TargetId$.inboundSchema,
        targetType: v2targettype_1.V2TargetType$.inboundSchema,
    });
    V2BulkElementAddMetadataData$.outboundSchema = z.object({
        metadata: z.record(z.string()),
        targetId: v2targetid_1.V2TargetId$.outboundSchema,
        targetType: v2targettype_1.V2TargetType$.outboundSchema,
    });
})(V2BulkElementAddMetadataData$ || (exports.V2BulkElementAddMetadataData$ = V2BulkElementAddMetadataData$ = {}));
/** @internal */
var V2BulkElementAddMetadata$;
(function (V2BulkElementAddMetadata$) {
    V2BulkElementAddMetadata$.inboundSchema = z.object({
        action: z.string(),
        data: z.lazy(() => V2BulkElementAddMetadataData$.inboundSchema).optional(),
        ik: z.string().optional(),
    });
    V2BulkElementAddMetadata$.outboundSchema = z.object({
        action: z.string(),
        data: z.lazy(() => V2BulkElementAddMetadataData$.outboundSchema).optional(),
        ik: z.string().optional(),
    });
})(V2BulkElementAddMetadata$ || (exports.V2BulkElementAddMetadata$ = V2BulkElementAddMetadata$ = {}));
