// @formance/formance-sdk/sdk/models/shared/v2bulkelementreverttransaction.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.V2BulkElementRevertTransaction$ = exports.V2BulkElementRevertTransactionData$ = void 0;
const z = __importStar(require("zod"));
/** @internal */
var V2BulkElementRevertTransactionData$;
(function (V2BulkElementRevertTransactionData$) {
    V2BulkElementRevertTransactionData$.inboundSchema = z.object({
        atEffectiveDate: z.boolean().optional(),
        force: z.boolean().optional(),
        id: z.number().transform((v) => BigInt(v)),
    });
    V2BulkElementRevertTransactionData$.outboundSchema = z.object({
        atEffectiveDate: z.boolean().optional(),
        force: z.boolean().optional(),
        id: z.bigint().transform((v) => Number(v)),
    });
})(V2BulkElementRevertTransactionData$ || (exports.V2BulkElementRevertTransactionData$ = V2BulkElementRevertTransactionData$ = {}));
/** @internal */
var V2BulkElementRevertTransaction$;
(function (V2BulkElementRevertTransaction$) {
    V2BulkElementRevertTransaction$.inboundSchema = z.object({
        action: z.string(),
        data: z.lazy(() => V2BulkElementRevertTransactionData$.inboundSchema).optional(),
        ik: z.string().optional(),
    });
    V2BulkElementRevertTransaction$.outboundSchema = z.object({
        action: z.string(),
        data: z.lazy(() => V2BulkElementRevertTransactionData$.outboundSchema).optional(),
        ik: z.string().optional(),
    });
})(V2BulkElementRevertTransaction$ || (exports.V2BulkElementRevertTransaction$ = V2BulkElementRevertTransaction$ = {}));
