// @formance/formance-sdk/sdk/models/shared/v2connector.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.V2Connector$ = exports.V2Connector = void 0;
const z = __importStar(require("zod"));
var V2Connector;
(function (V2Connector) {
    V2Connector["Stripe"] = "STRIPE";
    V2Connector["DummyPay"] = "DUMMY-PAY";
    V2Connector["Wise"] = "WISE";
    V2Connector["Modulr"] = "MODULR";
    V2Connector["CurrencyCloud"] = "CURRENCY-CLOUD";
    V2Connector["BankingCircle"] = "BANKING-CIRCLE";
    V2Connector["Mangopay"] = "MANGOPAY";
    V2Connector["Moneycorp"] = "MONEYCORP";
})(V2Connector || (exports.V2Connector = V2Connector = {}));
/** @internal */
var V2Connector$;
(function (V2Connector$) {
    V2Connector$.inboundSchema = z.nativeEnum(V2Connector);
    V2Connector$.outboundSchema = V2Connector$.inboundSchema;
})(V2Connector$ || (exports.V2Connector$ = V2Connector$ = {}));
