// @formance/formance-sdk/sdk/models/shared/v2debitwalletrequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.V2DebitWalletRequest$ = void 0;
const v2monetary_1 = require("./v2monetary");
const v2subject_1 = require("./v2subject");
const z = __importStar(require("zod"));
/** @internal */
var V2DebitWalletRequest$;
(function (V2DebitWalletRequest$) {
    V2DebitWalletRequest$.inboundSchema = z.object({
        amount: v2monetary_1.V2Monetary$.inboundSchema,
        balances: z.array(z.string()).optional(),
        description: z.string().optional(),
        destination: v2subject_1.V2Subject$.inboundSchema.optional(),
        metadata: z.record(z.string()),
        pending: z.boolean().optional(),
        timestamp: z
            .string()
            .datetime({ offset: true })
            .transform((v) => new Date(v))
            .optional(),
    });
    V2DebitWalletRequest$.outboundSchema = z.object({
        amount: v2monetary_1.V2Monetary$.outboundSchema,
        balances: z.array(z.string()).optional(),
        description: z.string().optional(),
        destination: v2subject_1.V2Subject$.outboundSchema.optional(),
        metadata: z.record(z.string()),
        pending: z.boolean().optional(),
        timestamp: z
            .date()
            .transform((v) => v.toISOString())
            .optional(),
    });
})(V2DebitWalletRequest$ || (exports.V2DebitWalletRequest$ = V2DebitWalletRequest$ = {}));
