// @formance/formance-sdk/sdk/models/shared/v2errorsenum.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.V2ErrorsEnum$ = exports.V2ErrorsEnum = void 0;
const z = __importStar(require("zod"));
var V2ErrorsEnum;
(function (V2ErrorsEnum) {
    V2ErrorsEnum["Internal"] = "INTERNAL";
    V2ErrorsEnum["InsufficientFund"] = "INSUFFICIENT_FUND";
    V2ErrorsEnum["Validation"] = "VALIDATION";
    V2ErrorsEnum["Conflict"] = "CONFLICT";
    V2ErrorsEnum["CompilationFailed"] = "COMPILATION_FAILED";
    V2ErrorsEnum["MetadataOverride"] = "METADATA_OVERRIDE";
    V2ErrorsEnum["NotFound"] = "NOT_FOUND";
    V2ErrorsEnum["RevertOccurring"] = "REVERT_OCCURRING";
    V2ErrorsEnum["AlreadyRevert"] = "ALREADY_REVERT";
    V2ErrorsEnum["NoPostings"] = "NO_POSTINGS";
    V2ErrorsEnum["LedgerNotFound"] = "LEDGER_NOT_FOUND";
})(V2ErrorsEnum || (exports.V2ErrorsEnum = V2ErrorsEnum = {}));
/** @internal */
var V2ErrorsEnum$;
(function (V2ErrorsEnum$) {
    V2ErrorsEnum$.inboundSchema = z.nativeEnum(V2ErrorsEnum);
    V2ErrorsEnum$.outboundSchema = V2ErrorsEnum$.inboundSchema;
})(V2ErrorsEnum$ || (exports.V2ErrorsEnum$ = V2ErrorsEnum$ = {}));
