// @formance/formance-sdk/sdk/models/shared/v2expandedtransaction.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.V2ExpandedTransaction$ = void 0;
const v2posting_1 = require("./v2posting");
const v2volume_1 = require("./v2volume");
const z = __importStar(require("zod"));
/** @internal */
var V2ExpandedTransaction$;
(function (V2ExpandedTransaction$) {
    V2ExpandedTransaction$.inboundSchema = z.object({
        id: z.number().transform((v) => BigInt(v)),
        metadata: z.record(z.string()),
        postCommitVolumes: z.record(z.record(v2volume_1.V2Volume$.inboundSchema)).optional(),
        postings: z.array(v2posting_1.V2Posting$.inboundSchema),
        preCommitVolumes: z.record(z.record(v2volume_1.V2Volume$.inboundSchema)).optional(),
        reference: z.string().optional(),
        reverted: z.boolean(),
        timestamp: z
            .string()
            .datetime({ offset: true })
            .transform((v) => new Date(v)),
    });
    V2ExpandedTransaction$.outboundSchema = z.object({
        id: z.bigint().transform((v) => Number(v)),
        metadata: z.record(z.string()),
        postCommitVolumes: z.record(z.record(v2volume_1.V2Volume$.outboundSchema)).optional(),
        postings: z.array(v2posting_1.V2Posting$.outboundSchema),
        preCommitVolumes: z.record(z.record(v2volume_1.V2Volume$.outboundSchema)).optional(),
        reference: z.string().optional(),
        reverted: z.boolean(),
        timestamp: z.date().transform((v) => v.toISOString()),
    });
})(V2ExpandedTransaction$ || (exports.V2ExpandedTransaction$ = V2ExpandedTransaction$ = {}));
