// @formance/formance-sdk/sdk/models/shared/v2hold.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.V2Hold$ = void 0;
const v2subject_1 = require("./v2subject");
const z = __importStar(require("zod"));
/** @internal */
var V2Hold$;
(function (V2Hold$) {
    V2Hold$.inboundSchema = z.object({
        description: z.string(),
        destination: v2subject_1.V2Subject$.inboundSchema.optional(),
        id: z.string(),
        metadata: z.record(z.string()),
        walletID: z.string(),
    });
    V2Hold$.outboundSchema = z.object({
        description: z.string(),
        destination: v2subject_1.V2Subject$.outboundSchema.optional(),
        id: z.string(),
        metadata: z.record(z.string()),
        walletID: z.string(),
    });
})(V2Hold$ || (exports.V2Hold$ = V2Hold$ = {}));
