// @formance/formance-sdk/sdk/models/shared/v2listtriggersresponse.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.V2ListTriggersResponse$ = exports.V2ListTriggersResponseCursor$ = void 0;
const v2trigger_1 = require("./v2trigger");
const z = __importStar(require("zod"));
/** @internal */
var V2ListTriggersResponseCursor$;
(function (V2ListTriggersResponseCursor$) {
    V2ListTriggersResponseCursor$.inboundSchema = z.object({
        data: z.array(v2trigger_1.V2Trigger$.inboundSchema),
        hasMore: z.boolean(),
        next: z.string().optional(),
        pageSize: z.number().int(),
        previous: z.string().optional(),
    });
    V2ListTriggersResponseCursor$.outboundSchema = z.object({
        data: z.array(v2trigger_1.V2Trigger$.outboundSchema),
        hasMore: z.boolean(),
        next: z.string().optional(),
        pageSize: z.number().int(),
        previous: z.string().optional(),
    });
})(V2ListTriggersResponseCursor$ || (exports.V2ListTriggersResponseCursor$ = V2ListTriggersResponseCursor$ = {}));
/** @internal */
var V2ListTriggersResponse$;
(function (V2ListTriggersResponse$) {
    V2ListTriggersResponse$.inboundSchema = z.object({
        cursor: z.lazy(() => V2ListTriggersResponseCursor$.inboundSchema),
    });
    V2ListTriggersResponse$.outboundSchema = z.object({
        cursor: z.lazy(() => V2ListTriggersResponseCursor$.outboundSchema),
    });
})(V2ListTriggersResponse$ || (exports.V2ListTriggersResponse$ = V2ListTriggersResponse$ = {}));
