// @formance/formance-sdk/sdk/models/shared/v2migrationinfo.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.V2MigrationInfo$ = exports.V2MigrationInfoState$ = exports.V2MigrationInfoState = void 0;
const z = __importStar(require("zod"));
var V2MigrationInfoState;
(function (V2MigrationInfoState) {
    V2MigrationInfoState["ToDo"] = "TO DO";
    V2MigrationInfoState["Done"] = "DONE";
})(V2MigrationInfoState || (exports.V2MigrationInfoState = V2MigrationInfoState = {}));
/** @internal */
var V2MigrationInfoState$;
(function (V2MigrationInfoState$) {
    V2MigrationInfoState$.inboundSchema = z.nativeEnum(V2MigrationInfoState);
    V2MigrationInfoState$.outboundSchema = V2MigrationInfoState$.inboundSchema;
})(V2MigrationInfoState$ || (exports.V2MigrationInfoState$ = V2MigrationInfoState$ = {}));
/** @internal */
var V2MigrationInfo$;
(function (V2MigrationInfo$) {
    V2MigrationInfo$.inboundSchema = z.object({
        date: z
            .string()
            .datetime({ offset: true })
            .transform((v) => new Date(v))
            .optional(),
        name: z.string().optional(),
        state: V2MigrationInfoState$.inboundSchema.optional(),
        version: z.number().int().optional(),
    });
    V2MigrationInfo$.outboundSchema = z.object({
        date: z
            .date()
            .transform((v) => v.toISOString())
            .optional(),
        name: z.string().optional(),
        state: V2MigrationInfoState$.outboundSchema.optional(),
        version: z.number().int().optional(),
    });
})(V2MigrationInfo$ || (exports.V2MigrationInfo$ = V2MigrationInfo$ = {}));
