// @formance/formance-sdk/sdk/models/shared/v2paymentadjustment.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.V2PaymentAdjustment$ = exports.V2PaymentAdjustmentRaw$ = void 0;
const v2paymentstatus_1 = require("./v2paymentstatus");
const z = __importStar(require("zod"));
/** @internal */
var V2PaymentAdjustmentRaw$;
(function (V2PaymentAdjustmentRaw$) {
    V2PaymentAdjustmentRaw$.inboundSchema = z.object({});
    V2PaymentAdjustmentRaw$.outboundSchema = z.object({});
})(V2PaymentAdjustmentRaw$ || (exports.V2PaymentAdjustmentRaw$ = V2PaymentAdjustmentRaw$ = {}));
/** @internal */
var V2PaymentAdjustment$;
(function (V2PaymentAdjustment$) {
    V2PaymentAdjustment$.inboundSchema = z.object({
        absolute: z.boolean(),
        amount: z.number().transform((v) => BigInt(v)),
        date: z
            .string()
            .datetime({ offset: true })
            .transform((v) => new Date(v)),
        raw: z.lazy(() => V2PaymentAdjustmentRaw$.inboundSchema),
        status: v2paymentstatus_1.V2PaymentStatus$.inboundSchema,
    });
    V2PaymentAdjustment$.outboundSchema = z.object({
        absolute: z.boolean(),
        amount: z.bigint().transform((v) => Number(v)),
        date: z.date().transform((v) => v.toISOString()),
        raw: z.lazy(() => V2PaymentAdjustmentRaw$.outboundSchema),
        status: v2paymentstatus_1.V2PaymentStatus$.outboundSchema,
    });
})(V2PaymentAdjustment$ || (exports.V2PaymentAdjustment$ = V2PaymentAdjustment$ = {}));
