// @formance/formance-sdk/sdk/models/shared/v2posttransaction.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.V2PostTransaction$ = exports.V2PostTransactionScript$ = void 0;
const v2posting_1 = require("./v2posting");
const z = __importStar(require("zod"));
/** @internal */
var V2PostTransactionScript$;
(function (V2PostTransactionScript$) {
    V2PostTransactionScript$.inboundSchema = z.object({
        plain: z.string(),
        vars: z.record(z.any()).optional(),
    });
    V2PostTransactionScript$.outboundSchema = z.object({
        plain: z.string(),
        vars: z.record(z.any()).optional(),
    });
})(V2PostTransactionScript$ || (exports.V2PostTransactionScript$ = V2PostTransactionScript$ = {}));
/** @internal */
var V2PostTransaction$;
(function (V2PostTransaction$) {
    V2PostTransaction$.inboundSchema = z.object({
        metadata: z.record(z.string()),
        postings: z.array(v2posting_1.V2Posting$.inboundSchema).optional(),
        reference: z.string().optional(),
        script: z.lazy(() => V2PostTransactionScript$.inboundSchema).optional(),
        timestamp: z
            .string()
            .datetime({ offset: true })
            .transform((v) => new Date(v))
            .optional(),
    });
    V2PostTransaction$.outboundSchema = z.object({
        metadata: z.record(z.string()),
        postings: z.array(v2posting_1.V2Posting$.outboundSchema).optional(),
        reference: z.string().optional(),
        script: z.lazy(() => V2PostTransactionScript$.outboundSchema).optional(),
        timestamp: z
            .date()
            .transform((v) => v.toISOString())
            .optional(),
    });
})(V2PostTransaction$ || (exports.V2PostTransaction$ = V2PostTransaction$ = {}));
