// @formance/formance-sdk/sdk/models/shared/v2transactionscursorresponse.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.V2TransactionsCursorResponse$ = exports.V2TransactionsCursorResponseCursor$ = void 0;
const v2expandedtransaction_1 = require("./v2expandedtransaction");
const z = __importStar(require("zod"));
/** @internal */
var V2TransactionsCursorResponseCursor$;
(function (V2TransactionsCursorResponseCursor$) {
    V2TransactionsCursorResponseCursor$.inboundSchema = z.object({
        data: z.array(v2expandedtransaction_1.V2ExpandedTransaction$.inboundSchema),
        hasMore: z.boolean(),
        next: z.string().optional(),
        pageSize: z.number().int(),
        previous: z.string().optional(),
    });
    V2TransactionsCursorResponseCursor$.outboundSchema = z.object({
        data: z.array(v2expandedtransaction_1.V2ExpandedTransaction$.outboundSchema),
        hasMore: z.boolean(),
        next: z.string().optional(),
        pageSize: z.number().int(),
        previous: z.string().optional(),
    });
})(V2TransactionsCursorResponseCursor$ || (exports.V2TransactionsCursorResponseCursor$ = V2TransactionsCursorResponseCursor$ = {}));
/** @internal */
var V2TransactionsCursorResponse$;
(function (V2TransactionsCursorResponse$) {
    V2TransactionsCursorResponse$.inboundSchema = z.object({
        cursor: z.lazy(() => V2TransactionsCursorResponseCursor$.inboundSchema),
    });
    V2TransactionsCursorResponse$.outboundSchema = z.object({
        cursor: z.lazy(() => V2TransactionsCursorResponseCursor$.outboundSchema),
    });
})(V2TransactionsCursorResponse$ || (exports.V2TransactionsCursorResponse$ = V2TransactionsCursorResponse$ = {}));
