// @formance/formance-sdk/sdk/models/shared/v2triggertest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.V2TriggerTest$ = exports.Variables$ = exports.Filter$ = void 0;
const z = __importStar(require("zod"));
/** @internal */
var Filter$;
(function (Filter$) {
    Filter$.inboundSchema = z.object({
        error: z.string().optional(),
        match: z.boolean().optional(),
    });
    Filter$.outboundSchema = z.object({
        error: z.string().optional(),
        match: z.boolean().optional(),
    });
})(Filter$ || (exports.Filter$ = Filter$ = {}));
/** @internal */
var Variables$;
(function (Variables$) {
    Variables$.inboundSchema = z.object({
        error: z.string().optional(),
        value: z.string().optional(),
    });
    Variables$.outboundSchema = z.object({
        error: z.string().optional(),
        value: z.string().optional(),
    });
})(Variables$ || (exports.Variables$ = Variables$ = {}));
/** @internal */
var V2TriggerTest$;
(function (V2TriggerTest$) {
    V2TriggerTest$.inboundSchema = z.object({
        filter: z.lazy(() => Filter$.inboundSchema).optional(),
        variables: z.record(z.lazy(() => Variables$.inboundSchema)).optional(),
    });
    V2TriggerTest$.outboundSchema = z.object({
        filter: z.lazy(() => Filter$.outboundSchema).optional(),
        variables: z.record(z.lazy(() => Variables$.outboundSchema)).optional(),
    });
})(V2TriggerTest$ || (exports.V2TriggerTest$ = V2TriggerTest$ = {}));
