// @formance/formance-sdk/sdk/models/shared/v2triggeroccurrence.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.V2TriggerOccurrence$ = void 0;
const v2workflowinstance_1 = require("./v2workflowinstance");
const z = __importStar(require("zod"));
/** @internal */
var V2TriggerOccurrence$;
(function (V2TriggerOccurrence$) {
    V2TriggerOccurrence$.inboundSchema = z.object({
        date: z
            .string()
            .datetime({ offset: true })
            .transform((v) => new Date(v)),
        error: z.string().optional(),
        event: z.record(z.any()),
        triggerID: z.string(),
        workflowInstance: v2workflowinstance_1.V2WorkflowInstance$.inboundSchema.optional(),
        workflowInstanceID: z.string().optional(),
    });
    V2TriggerOccurrence$.outboundSchema = z.object({
        date: z.date().transform((v) => v.toISOString()),
        error: z.string().optional(),
        event: z.record(z.any()),
        triggerID: z.string(),
        workflowInstance: v2workflowinstance_1.V2WorkflowInstance$.outboundSchema.optional(),
        workflowInstanceID: z.string().optional(),
    });
})(V2TriggerOccurrence$ || (exports.V2TriggerOccurrence$ = V2TriggerOccurrence$ = {}));
