// @formance/formance-sdk/sdk/models/shared/v2volumeswithbalancecursorresponse.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.V2VolumesWithBalanceCursorResponse$ = exports.V2VolumesWithBalanceCursorResponseCursor$ = void 0;
const v2volumeswithbalance_1 = require("./v2volumeswithbalance");
const z = __importStar(require("zod"));
/** @internal */
var V2VolumesWithBalanceCursorResponseCursor$;
(function (V2VolumesWithBalanceCursorResponseCursor$) {
    V2VolumesWithBalanceCursorResponseCursor$.inboundSchema = z.object({
        data: z.array(v2volumeswithbalance_1.V2VolumesWithBalance$.inboundSchema),
        hasMore: z.boolean(),
        next: z.string().optional(),
        pageSize: z.number().int(),
        previous: z.string().optional(),
    });
    V2VolumesWithBalanceCursorResponseCursor$.outboundSchema = z.object({
        data: z.array(v2volumeswithbalance_1.V2VolumesWithBalance$.outboundSchema),
        hasMore: z.boolean(),
        next: z.string().optional(),
        pageSize: z.number().int(),
        previous: z.string().optional(),
    });
})(V2VolumesWithBalanceCursorResponseCursor$ || (exports.V2VolumesWithBalanceCursorResponseCursor$ = V2VolumesWithBalanceCursorResponseCursor$ = {}));
/** @internal */
var V2VolumesWithBalanceCursorResponse$;
(function (V2VolumesWithBalanceCursorResponse$) {
    V2VolumesWithBalanceCursorResponse$.inboundSchema = z.object({
        cursor: z.lazy(() => V2VolumesWithBalanceCursorResponseCursor$.inboundSchema),
    });
    V2VolumesWithBalanceCursorResponse$.outboundSchema = z.object({
        cursor: z.lazy(() => V2VolumesWithBalanceCursorResponseCursor$.outboundSchema),
    });
})(V2VolumesWithBalanceCursorResponse$ || (exports.V2VolumesWithBalanceCursorResponse$ = V2VolumesWithBalanceCursorResponse$ = {}));
