// @formance/formance-sdk/sdk/models/shared/v2walletwithbalances.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.V2WalletWithBalances$ = exports.Balances$ = void 0;
const v2assetholder_1 = require("./v2assetholder");
const z = __importStar(require("zod"));
/** @internal */
var Balances$;
(function (Balances$) {
    Balances$.inboundSchema = z.object({
        main: v2assetholder_1.V2AssetHolder$.inboundSchema,
    });
    Balances$.outboundSchema = z.object({
        main: v2assetholder_1.V2AssetHolder$.outboundSchema,
    });
})(Balances$ || (exports.Balances$ = Balances$ = {}));
/** @internal */
var V2WalletWithBalances$;
(function (V2WalletWithBalances$) {
    V2WalletWithBalances$.inboundSchema = z.object({
        balances: z.lazy(() => Balances$.inboundSchema),
        createdAt: z
            .string()
            .datetime({ offset: true })
            .transform((v) => new Date(v)),
        id: z.string(),
        ledger: z.string(),
        metadata: z.record(z.string()),
        name: z.string(),
    });
    V2WalletWithBalances$.outboundSchema = z.object({
        balances: z.lazy(() => Balances$.outboundSchema),
        createdAt: z.date().transform((v) => v.toISOString()),
        id: z.string(),
        ledger: z.string(),
        metadata: z.record(z.string()),
        name: z.string(),
    });
})(V2WalletWithBalances$ || (exports.V2WalletWithBalances$ = V2WalletWithBalances$ = {}));
