// @formance/formance-sdk/sdk/models/shared/v2workflowinstancehistorystage.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.V2WorkflowInstanceHistoryStage$ = void 0;
const v2workflowinstancehistorystageinput_1 = require("./v2workflowinstancehistorystageinput");
const v2workflowinstancehistorystageoutput_1 = require("./v2workflowinstancehistorystageoutput");
const z = __importStar(require("zod"));
/** @internal */
var V2WorkflowInstanceHistoryStage$;
(function (V2WorkflowInstanceHistoryStage$) {
    V2WorkflowInstanceHistoryStage$.inboundSchema = z.object({
        attempt: z.number().int(),
        error: z.string().optional(),
        input: v2workflowinstancehistorystageinput_1.V2WorkflowInstanceHistoryStageInput$.inboundSchema,
        lastFailure: z.string().optional(),
        name: z.string(),
        nextExecution: z
            .string()
            .datetime({ offset: true })
            .transform((v) => new Date(v))
            .optional(),
        output: v2workflowinstancehistorystageoutput_1.V2WorkflowInstanceHistoryStageOutput$.inboundSchema.optional(),
        startedAt: z
            .string()
            .datetime({ offset: true })
            .transform((v) => new Date(v)),
        terminated: z.boolean(),
        terminatedAt: z
            .string()
            .datetime({ offset: true })
            .transform((v) => new Date(v))
            .optional(),
    });
    V2WorkflowInstanceHistoryStage$.outboundSchema = z.object({
        attempt: z.number().int(),
        error: z.string().optional(),
        input: v2workflowinstancehistorystageinput_1.V2WorkflowInstanceHistoryStageInput$.outboundSchema,
        lastFailure: z.string().optional(),
        name: z.string(),
        nextExecution: z
            .date()
            .transform((v) => v.toISOString())
            .optional(),
        output: v2workflowinstancehistorystageoutput_1.V2WorkflowInstanceHistoryStageOutput$.outboundSchema.optional(),
        startedAt: z.date().transform((v) => v.toISOString()),
        terminated: z.boolean(),
        terminatedAt: z
            .date()
            .transform((v) => v.toISOString())
            .optional(),
    });
})(V2WorkflowInstanceHistoryStage$ || (exports.V2WorkflowInstanceHistoryStage$ = V2WorkflowInstanceHistoryStage$ = {}));
