// @formance/formance-sdk/sdk/models/shared/v2workflowinstance.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.V2WorkflowInstance$ = void 0;
const v2stagestatus_1 = require("./v2stagestatus");
const z = __importStar(require("zod"));
/** @internal */
var V2WorkflowInstance$;
(function (V2WorkflowInstance$) {
    V2WorkflowInstance$.inboundSchema = z.object({
        createdAt: z
            .string()
            .datetime({ offset: true })
            .transform((v) => new Date(v)),
        error: z.string().optional(),
        id: z.string(),
        status: z.array(v2stagestatus_1.V2StageStatus$.inboundSchema).optional(),
        terminated: z.boolean(),
        terminatedAt: z
            .string()
            .datetime({ offset: true })
            .transform((v) => new Date(v))
            .optional(),
        updatedAt: z
            .string()
            .datetime({ offset: true })
            .transform((v) => new Date(v)),
        workflowID: z.string(),
    });
    V2WorkflowInstance$.outboundSchema = z.object({
        createdAt: z.date().transform((v) => v.toISOString()),
        error: z.string().optional(),
        id: z.string(),
        status: z.array(v2stagestatus_1.V2StageStatus$.outboundSchema).optional(),
        terminated: z.boolean(),
        terminatedAt: z
            .date()
            .transform((v) => v.toISOString())
            .optional(),
        updatedAt: z.date().transform((v) => v.toISOString()),
        workflowID: z.string(),
    });
})(V2WorkflowInstance$ || (exports.V2WorkflowInstance$ = V2WorkflowInstance$ = {}));
