// @formance/formance-sdk/sdk/models/shared/v2workflowinstancehistorystageoutput.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.V2WorkflowInstanceHistoryStageOutput$ = void 0;
const primitives_1 = require("../../../lib/primitives");
const v2activitycreatetransactionoutput_1 = require("./v2activitycreatetransactionoutput");
const v2activitydebitwalletoutput_1 = require("./v2activitydebitwalletoutput");
const v2activitygetaccountoutput_1 = require("./v2activitygetaccountoutput");
const v2activitygetpaymentoutput_1 = require("./v2activitygetpaymentoutput");
const v2activitygetwalletoutput_1 = require("./v2activitygetwalletoutput");
const v2listwalletsresponse_1 = require("./v2listwalletsresponse");
const z = __importStar(require("zod"));
/** @internal */
var V2WorkflowInstanceHistoryStageOutput$;
(function (V2WorkflowInstanceHistoryStageOutput$) {
    V2WorkflowInstanceHistoryStageOutput$.inboundSchema = z
        .object({
        CreateTransaction: v2activitycreatetransactionoutput_1.V2ActivityCreateTransactionOutput$.inboundSchema.optional(),
        DebitWallet: v2activitydebitwalletoutput_1.V2ActivityDebitWalletOutput$.inboundSchema.optional(),
        GetAccount: v2activitygetaccountoutput_1.V2ActivityGetAccountOutput$.inboundSchema.optional(),
        GetPayment: v2activitygetpaymentoutput_1.V2ActivityGetPaymentOutput$.inboundSchema.optional(),
        GetWallet: v2activitygetwalletoutput_1.V2ActivityGetWalletOutput$.inboundSchema.optional(),
        ListWallets: v2listwalletsresponse_1.V2ListWalletsResponse$.inboundSchema.optional(),
    })
        .transform((v) => {
        return (0, primitives_1.remap)(v, {
            CreateTransaction: "createTransaction",
            DebitWallet: "debitWallet",
            GetAccount: "getAccount",
            GetPayment: "getPayment",
            GetWallet: "getWallet",
            ListWallets: "listWallets",
        });
    });
    V2WorkflowInstanceHistoryStageOutput$.outboundSchema = z
        .object({
        createTransaction: v2activitycreatetransactionoutput_1.V2ActivityCreateTransactionOutput$.outboundSchema.optional(),
        debitWallet: v2activitydebitwalletoutput_1.V2ActivityDebitWalletOutput$.outboundSchema.optional(),
        getAccount: v2activitygetaccountoutput_1.V2ActivityGetAccountOutput$.outboundSchema.optional(),
        getPayment: v2activitygetpaymentoutput_1.V2ActivityGetPaymentOutput$.outboundSchema.optional(),
        getWallet: v2activitygetwalletoutput_1.V2ActivityGetWalletOutput$.outboundSchema.optional(),
        listWallets: v2listwalletsresponse_1.V2ListWalletsResponse$.outboundSchema.optional(),
    })
        .transform((v) => {
        return (0, primitives_1.remap)(v, {
            createTransaction: "CreateTransaction",
            debitWallet: "DebitWallet",
            getAccount: "GetAccount",
            getPayment: "GetPayment",
            getWallet: "GetWallet",
            listWallets: "ListWallets",
        });
    });
})(V2WorkflowInstanceHistoryStageOutput$ || (exports.V2WorkflowInstanceHistoryStageOutput$ = V2WorkflowInstanceHistoryStageOutput$ = {}));
