// @formance/formance-sdk/sdk/models/shared/workflowinstancehistory.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkflowInstanceHistory$ = void 0;
const stage_1 = require("./stage");
const z = __importStar(require("zod"));
/** @internal */
var WorkflowInstanceHistory$;
(function (WorkflowInstanceHistory$) {
    WorkflowInstanceHistory$.inboundSchema = z.object({
        error: z.string().optional(),
        input: stage_1.Stage$.inboundSchema,
        name: z.string(),
        startedAt: z
            .string()
            .datetime({ offset: true })
            .transform((v) => new Date(v)),
        terminated: z.boolean(),
        terminatedAt: z
            .string()
            .datetime({ offset: true })
            .transform((v) => new Date(v))
            .optional(),
    });
    WorkflowInstanceHistory$.outboundSchema = z.object({
        error: z.string().optional(),
        input: stage_1.Stage$.outboundSchema,
        name: z.string(),
        startedAt: z.date().transform((v) => v.toISOString()),
        terminated: z.boolean(),
        terminatedAt: z
            .date()
            .transform((v) => v.toISOString())
            .optional(),
    });
})(WorkflowInstanceHistory$ || (exports.WorkflowInstanceHistory$ = WorkflowInstanceHistory$ = {}));
