// @formance/formance-sdk/sdk/models/operations/installconnector.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.InstallConnectorResponse$ = exports.InstallConnectorRequest$ = void 0;
const primitives_1 = require("../../../lib/primitives");
const shared = __importStar(require("../shared"));
const z = __importStar(require("zod"));
/** @internal */
var InstallConnectorRequest$;
(function (InstallConnectorRequest$) {
    InstallConnectorRequest$.inboundSchema = z
        .object({
        ConnectorConfig: shared.ConnectorConfig$.inboundSchema,
        connector: shared.Connector$.inboundSchema,
    })
        .transform((v) => {
        return (0, primitives_1.remap)(v, {
            ConnectorConfig: "connectorConfig",
        });
    });
    InstallConnectorRequest$.outboundSchema = z
        .object({
        connectorConfig: shared.ConnectorConfig$.outboundSchema,
        connector: shared.Connector$.outboundSchema,
    })
        .transform((v) => {
        return (0, primitives_1.remap)(v, {
            connectorConfig: "ConnectorConfig",
        });
    });
})(InstallConnectorRequest$ || (exports.InstallConnectorRequest$ = InstallConnectorRequest$ = {}));
/** @internal */
var InstallConnectorResponse$;
(function (InstallConnectorResponse$) {
    InstallConnectorResponse$.inboundSchema = z
        .object({
        ConnectorResponse: shared.ConnectorResponse$.inboundSchema.optional(),
        ContentType: z.string(),
        StatusCode: z.number().int(),
        RawResponse: z.instanceof(Response),
    })
        .transform((v) => {
        return (0, primitives_1.remap)(v, {
            ConnectorResponse: "connectorResponse",
            ContentType: "contentType",
            StatusCode: "statusCode",
            RawResponse: "rawResponse",
        });
    });
    InstallConnectorResponse$.outboundSchema = z
        .object({
        connectorResponse: shared.ConnectorResponse$.outboundSchema.optional(),
        contentType: z.string(),
        statusCode: z.number().int(),
        rawResponse: z.instanceof(Response).transform(() => {
            throw new Error("Response cannot be serialized");
        }),
    })
        .transform((v) => {
        return (0, primitives_1.remap)(v, {
            connectorResponse: "ConnectorResponse",
            contentType: "ContentType",
            statusCode: "StatusCode",
            rawResponse: "RawResponse",
        });
    });
})(InstallConnectorResponse$ || (exports.InstallConnectorResponse$ = InstallConnectorResponse$ = {}));
