// @formance/formance-sdk/sdk/models/operations/listtransactions.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListTransactionsResponse$ = exports.ListTransactionsRequest$ = void 0;
const primitives_1 = require("../../../lib/primitives");
const shared = __importStar(require("../shared"));
const z = __importStar(require("zod"));
/** @internal */
var ListTransactionsRequest$;
(function (ListTransactionsRequest$) {
    ListTransactionsRequest$.inboundSchema = z.object({
        account: z.string().optional(),
        after: z.string().optional(),
        cursor: z.string().optional(),
        destination: z.string().optional(),
        endTime: z
            .string()
            .datetime({ offset: true })
            .transform((v) => new Date(v))
            .optional(),
        ledger: z.string(),
        metadata: z.record(z.any()).optional(),
        pageSize: z.number().int().default(15),
        reference: z.string().optional(),
        source: z.string().optional(),
        startTime: z
            .string()
            .datetime({ offset: true })
            .transform((v) => new Date(v))
            .optional(),
    });
    ListTransactionsRequest$.outboundSchema = z.object({
        account: z.string().optional(),
        after: z.string().optional(),
        cursor: z.string().optional(),
        destination: z.string().optional(),
        endTime: z
            .date()
            .transform((v) => v.toISOString())
            .optional(),
        ledger: z.string(),
        metadata: z.record(z.any()).optional(),
        pageSize: z.number().int().default(15),
        reference: z.string().optional(),
        source: z.string().optional(),
        startTime: z
            .date()
            .transform((v) => v.toISOString())
            .optional(),
    });
})(ListTransactionsRequest$ || (exports.ListTransactionsRequest$ = ListTransactionsRequest$ = {}));
/** @internal */
var ListTransactionsResponse$;
(function (ListTransactionsResponse$) {
    ListTransactionsResponse$.inboundSchema = z
        .object({
        ContentType: z.string(),
        StatusCode: z.number().int(),
        RawResponse: z.instanceof(Response),
        TransactionsCursorResponse: shared.TransactionsCursorResponse$.inboundSchema.optional(),
    })
        .transform((v) => {
        return (0, primitives_1.remap)(v, {
            ContentType: "contentType",
            StatusCode: "statusCode",
            RawResponse: "rawResponse",
            TransactionsCursorResponse: "transactionsCursorResponse",
        });
    });
    ListTransactionsResponse$.outboundSchema = z
        .object({
        contentType: z.string(),
        statusCode: z.number().int(),
        rawResponse: z.instanceof(Response).transform(() => {
            throw new Error("Response cannot be serialized");
        }),
        transactionsCursorResponse: shared.TransactionsCursorResponse$.outboundSchema.optional(),
    })
        .transform((v) => {
        return (0, primitives_1.remap)(v, {
            contentType: "ContentType",
            statusCode: "StatusCode",
            rawResponse: "RawResponse",
            transactionsCursorResponse: "TransactionsCursorResponse",
        });
    });
})(ListTransactionsResponse$ || (exports.ListTransactionsResponse$ = ListTransactionsResponse$ = {}));
