// @formance/formance-sdk/sdk/models/operations/runworkflow.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RunWorkflowResponse$ = exports.RunWorkflowRequest$ = void 0;
const primitives_1 = require("../../../lib/primitives");
const shared = __importStar(require("../shared"));
const z = __importStar(require("zod"));
/** @internal */
var RunWorkflowRequest$;
(function (RunWorkflowRequest$) {
    RunWorkflowRequest$.inboundSchema = z
        .object({
        RequestBody: z.record(z.string()).optional(),
        wait: z.boolean().optional(),
        workflowID: z.string(),
    })
        .transform((v) => {
        return (0, primitives_1.remap)(v, {
            RequestBody: "requestBody",
        });
    });
    RunWorkflowRequest$.outboundSchema = z
        .object({
        requestBody: z.record(z.string()).optional(),
        wait: z.boolean().optional(),
        workflowID: z.string(),
    })
        .transform((v) => {
        return (0, primitives_1.remap)(v, {
            requestBody: "RequestBody",
        });
    });
})(RunWorkflowRequest$ || (exports.RunWorkflowRequest$ = RunWorkflowRequest$ = {}));
/** @internal */
var RunWorkflowResponse$;
(function (RunWorkflowResponse$) {
    RunWorkflowResponse$.inboundSchema = z
        .object({
        ContentType: z.string(),
        RunWorkflowResponse: shared.RunWorkflowResponse$.inboundSchema.optional(),
        StatusCode: z.number().int(),
        RawResponse: z.instanceof(Response),
    })
        .transform((v) => {
        return (0, primitives_1.remap)(v, {
            ContentType: "contentType",
            RunWorkflowResponse: "runWorkflowResponse",
            StatusCode: "statusCode",
            RawResponse: "rawResponse",
        });
    });
    RunWorkflowResponse$.outboundSchema = z
        .object({
        contentType: z.string(),
        runWorkflowResponse: shared.RunWorkflowResponse$.outboundSchema.optional(),
        statusCode: z.number().int(),
        rawResponse: z.instanceof(Response).transform(() => {
            throw new Error("Response cannot be serialized");
        }),
    })
        .transform((v) => {
        return (0, primitives_1.remap)(v, {
            contentType: "ContentType",
            runWorkflowResponse: "RunWorkflowResponse",
            statusCode: "StatusCode",
            rawResponse: "RawResponse",
        });
    });
})(RunWorkflowResponse$ || (exports.RunWorkflowResponse$ = RunWorkflowResponse$ = {}));
