// @formance/formance-sdk/sdk/models/operations/v2addmetadataontransaction.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.V2AddMetadataOnTransactionResponse$ = exports.V2AddMetadataOnTransactionRequest$ = void 0;
const primitives_1 = require("../../../lib/primitives");
const z = __importStar(require("zod"));
/** @internal */
var V2AddMetadataOnTransactionRequest$;
(function (V2AddMetadataOnTransactionRequest$) {
    V2AddMetadataOnTransactionRequest$.inboundSchema = z
        .object({
        "Idempotency-Key": z.string().optional(),
        RequestBody: z.record(z.string()).optional(),
        dryRun: z.boolean().optional(),
        id: z.number().transform((v) => BigInt(v)),
        ledger: z.string(),
    })
        .transform((v) => {
        return (0, primitives_1.remap)(v, {
            "Idempotency-Key": "idempotencyKey",
            RequestBody: "requestBody",
        });
    });
    V2AddMetadataOnTransactionRequest$.outboundSchema = z
        .object({
        idempotencyKey: z.string().optional(),
        requestBody: z.record(z.string()).optional(),
        dryRun: z.boolean().optional(),
        id: z.bigint().transform((v) => Number(v)),
        ledger: z.string(),
    })
        .transform((v) => {
        return (0, primitives_1.remap)(v, {
            idempotencyKey: "Idempotency-Key",
            requestBody: "RequestBody",
        });
    });
})(V2AddMetadataOnTransactionRequest$ || (exports.V2AddMetadataOnTransactionRequest$ = V2AddMetadataOnTransactionRequest$ = {}));
/** @internal */
var V2AddMetadataOnTransactionResponse$;
(function (V2AddMetadataOnTransactionResponse$) {
    V2AddMetadataOnTransactionResponse$.inboundSchema = z
        .object({
        ContentType: z.string(),
        StatusCode: z.number().int(),
        RawResponse: z.instanceof(Response),
    })
        .transform((v) => {
        return (0, primitives_1.remap)(v, {
            ContentType: "contentType",
            StatusCode: "statusCode",
            RawResponse: "rawResponse",
        });
    });
    V2AddMetadataOnTransactionResponse$.outboundSchema = z
        .object({
        contentType: z.string(),
        statusCode: z.number().int(),
        rawResponse: z.instanceof(Response).transform(() => {
            throw new Error("Response cannot be serialized");
        }),
    })
        .transform((v) => {
        return (0, primitives_1.remap)(v, {
            contentType: "ContentType",
            statusCode: "StatusCode",
            rawResponse: "RawResponse",
        });
    });
})(V2AddMetadataOnTransactionResponse$ || (exports.V2AddMetadataOnTransactionResponse$ = V2AddMetadataOnTransactionResponse$ = {}));
