// @formance/formance-sdk/sdk/models/operations/v2counttransactions.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.V2CountTransactionsResponse$ = exports.V2CountTransactionsRequest$ = void 0;
const primitives_1 = require("../../../lib/primitives");
const z = __importStar(require("zod"));
/** @internal */
var V2CountTransactionsRequest$;
(function (V2CountTransactionsRequest$) {
    V2CountTransactionsRequest$.inboundSchema = z.object({
        ledger: z.string(),
        pit: z
            .string()
            .datetime({ offset: true })
            .transform((v) => new Date(v))
            .optional(),
        query: z.record(z.any()).optional(),
    });
    V2CountTransactionsRequest$.outboundSchema = z.object({
        ledger: z.string(),
        pit: z
            .date()
            .transform((v) => v.toISOString())
            .optional(),
        query: z.record(z.any()).optional(),
    });
})(V2CountTransactionsRequest$ || (exports.V2CountTransactionsRequest$ = V2CountTransactionsRequest$ = {}));
/** @internal */
var V2CountTransactionsResponse$;
(function (V2CountTransactionsResponse$) {
    V2CountTransactionsResponse$.inboundSchema = z
        .object({
        ContentType: z.string(),
        Headers: z.record(z.array(z.string())),
        StatusCode: z.number().int(),
        RawResponse: z.instanceof(Response),
    })
        .transform((v) => {
        return (0, primitives_1.remap)(v, {
            ContentType: "contentType",
            Headers: "headers",
            StatusCode: "statusCode",
            RawResponse: "rawResponse",
        });
    });
    V2CountTransactionsResponse$.outboundSchema = z
        .object({
        contentType: z.string(),
        headers: z.record(z.array(z.string())),
        statusCode: z.number().int(),
        rawResponse: z.instanceof(Response).transform(() => {
            throw new Error("Response cannot be serialized");
        }),
    })
        .transform((v) => {
        return (0, primitives_1.remap)(v, {
            contentType: "ContentType",
            headers: "Headers",
            statusCode: "StatusCode",
            rawResponse: "RawResponse",
        });
    });
})(V2CountTransactionsResponse$ || (exports.V2CountTransactionsResponse$ = V2CountTransactionsResponse$ = {}));
