// @formance/formance-sdk/sdk/models/operations/v2createtransaction.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.V2CreateTransactionResponse$ = exports.V2CreateTransactionRequest$ = void 0;
const primitives_1 = require("../../../lib/primitives");
const shared = __importStar(require("../shared"));
const z = __importStar(require("zod"));
/** @internal */
var V2CreateTransactionRequest$;
(function (V2CreateTransactionRequest$) {
    V2CreateTransactionRequest$.inboundSchema = z
        .object({
        "Idempotency-Key": z.string().optional(),
        V2PostTransaction: shared.V2PostTransaction$.inboundSchema,
        dryRun: z.boolean().optional(),
        ledger: z.string(),
    })
        .transform((v) => {
        return (0, primitives_1.remap)(v, {
            "Idempotency-Key": "idempotencyKey",
            V2PostTransaction: "v2PostTransaction",
        });
    });
    V2CreateTransactionRequest$.outboundSchema = z
        .object({
        idempotencyKey: z.string().optional(),
        v2PostTransaction: shared.V2PostTransaction$.outboundSchema,
        dryRun: z.boolean().optional(),
        ledger: z.string(),
    })
        .transform((v) => {
        return (0, primitives_1.remap)(v, {
            idempotencyKey: "Idempotency-Key",
            v2PostTransaction: "V2PostTransaction",
        });
    });
})(V2CreateTransactionRequest$ || (exports.V2CreateTransactionRequest$ = V2CreateTransactionRequest$ = {}));
/** @internal */
var V2CreateTransactionResponse$;
(function (V2CreateTransactionResponse$) {
    V2CreateTransactionResponse$.inboundSchema = z
        .object({
        ContentType: z.string(),
        StatusCode: z.number().int(),
        RawResponse: z.instanceof(Response),
        V2CreateTransactionResponse: shared.V2CreateTransactionResponse$.inboundSchema.optional(),
    })
        .transform((v) => {
        return (0, primitives_1.remap)(v, {
            ContentType: "contentType",
            StatusCode: "statusCode",
            RawResponse: "rawResponse",
            V2CreateTransactionResponse: "v2CreateTransactionResponse",
        });
    });
    V2CreateTransactionResponse$.outboundSchema = z
        .object({
        contentType: z.string(),
        statusCode: z.number().int(),
        rawResponse: z.instanceof(Response).transform(() => {
            throw new Error("Response cannot be serialized");
        }),
        v2CreateTransactionResponse: shared.V2CreateTransactionResponse$.outboundSchema.optional(),
    })
        .transform((v) => {
        return (0, primitives_1.remap)(v, {
            contentType: "ContentType",
            statusCode: "StatusCode",
            rawResponse: "RawResponse",
            v2CreateTransactionResponse: "V2CreateTransactionResponse",
        });
    });
})(V2CreateTransactionResponse$ || (exports.V2CreateTransactionResponse$ = V2CreateTransactionResponse$ = {}));
