// @formance/formance-sdk/sdk/models/operations/v2createbulk.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.V2CreateBulkResponse$ = exports.V2CreateBulkRequest$ = void 0;
const primitives_1 = require("../../../lib/primitives");
const shared = __importStar(require("../shared"));
const z = __importStar(require("zod"));
/** @internal */
var V2CreateBulkRequest$;
(function (V2CreateBulkRequest$) {
    V2CreateBulkRequest$.inboundSchema = z
        .object({
        RequestBody: z.array(shared.V2BulkElement$.inboundSchema).optional(),
        ledger: z.string(),
    })
        .transform((v) => {
        return (0, primitives_1.remap)(v, {
            RequestBody: "requestBody",
        });
    });
    V2CreateBulkRequest$.outboundSchema = z
        .object({
        requestBody: z.array(shared.V2BulkElement$.outboundSchema).optional(),
        ledger: z.string(),
    })
        .transform((v) => {
        return (0, primitives_1.remap)(v, {
            requestBody: "RequestBody",
        });
    });
})(V2CreateBulkRequest$ || (exports.V2CreateBulkRequest$ = V2CreateBulkRequest$ = {}));
/** @internal */
var V2CreateBulkResponse$;
(function (V2CreateBulkResponse$) {
    V2CreateBulkResponse$.inboundSchema = z
        .object({
        ContentType: z.string(),
        StatusCode: z.number().int(),
        RawResponse: z.instanceof(Response),
        V2BulkResponse: shared.V2BulkResponse$.inboundSchema.optional(),
    })
        .transform((v) => {
        return (0, primitives_1.remap)(v, {
            ContentType: "contentType",
            StatusCode: "statusCode",
            RawResponse: "rawResponse",
            V2BulkResponse: "v2BulkResponse",
        });
    });
    V2CreateBulkResponse$.outboundSchema = z
        .object({
        contentType: z.string(),
        statusCode: z.number().int(),
        rawResponse: z.instanceof(Response).transform(() => {
            throw new Error("Response cannot be serialized");
        }),
        v2BulkResponse: shared.V2BulkResponse$.outboundSchema.optional(),
    })
        .transform((v) => {
        return (0, primitives_1.remap)(v, {
            contentType: "ContentType",
            statusCode: "StatusCode",
            rawResponse: "RawResponse",
            v2BulkResponse: "V2BulkResponse",
        });
    });
})(V2CreateBulkResponse$ || (exports.V2CreateBulkResponse$ = V2CreateBulkResponse$ = {}));
