// @formance/formance-sdk/sdk/models/operations/v2getinstancestagehistory.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.V2GetInstanceStageHistoryResponse$ = exports.V2GetInstanceStageHistoryRequest$ = void 0;
const primitives_1 = require("../../../lib/primitives");
const shared = __importStar(require("../shared"));
const z = __importStar(require("zod"));
/** @internal */
var V2GetInstanceStageHistoryRequest$;
(function (V2GetInstanceStageHistoryRequest$) {
    V2GetInstanceStageHistoryRequest$.inboundSchema = z.object({
        instanceID: z.string(),
        number: z.number().int(),
    });
    V2GetInstanceStageHistoryRequest$.outboundSchema = z.object({
        instanceID: z.string(),
        number: z.number().int(),
    });
})(V2GetInstanceStageHistoryRequest$ || (exports.V2GetInstanceStageHistoryRequest$ = V2GetInstanceStageHistoryRequest$ = {}));
/** @internal */
var V2GetInstanceStageHistoryResponse$;
(function (V2GetInstanceStageHistoryResponse$) {
    V2GetInstanceStageHistoryResponse$.inboundSchema = z
        .object({
        ContentType: z.string(),
        StatusCode: z.number().int(),
        RawResponse: z.instanceof(Response),
        V2GetWorkflowInstanceHistoryStageResponse: shared.V2GetWorkflowInstanceHistoryStageResponse$.inboundSchema.optional(),
    })
        .transform((v) => {
        return (0, primitives_1.remap)(v, {
            ContentType: "contentType",
            StatusCode: "statusCode",
            RawResponse: "rawResponse",
            V2GetWorkflowInstanceHistoryStageResponse: "v2GetWorkflowInstanceHistoryStageResponse",
        });
    });
    V2GetInstanceStageHistoryResponse$.outboundSchema = z
        .object({
        contentType: z.string(),
        statusCode: z.number().int(),
        rawResponse: z.instanceof(Response).transform(() => {
            throw new Error("Response cannot be serialized");
        }),
        v2GetWorkflowInstanceHistoryStageResponse: shared.V2GetWorkflowInstanceHistoryStageResponse$.outboundSchema.optional(),
    })
        .transform((v) => {
        return (0, primitives_1.remap)(v, {
            contentType: "ContentType",
            statusCode: "StatusCode",
            rawResponse: "RawResponse",
            v2GetWorkflowInstanceHistoryStageResponse: "V2GetWorkflowInstanceHistoryStageResponse",
        });
    });
})(V2GetInstanceStageHistoryResponse$ || (exports.V2GetInstanceStageHistoryResponse$ = V2GetInstanceStageHistoryResponse$ = {}));
