// @formance/formance-sdk/sdk/models/operations/v2getvolumeswithbalances.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.V2GetVolumesWithBalancesResponse$ = exports.V2GetVolumesWithBalancesRequest$ = void 0;
const primitives_1 = require("../../../lib/primitives");
const shared = __importStar(require("../shared"));
const z = __importStar(require("zod"));
/** @internal */
var V2GetVolumesWithBalancesRequest$;
(function (V2GetVolumesWithBalancesRequest$) {
    V2GetVolumesWithBalancesRequest$.inboundSchema = z.object({
        cursor: z.string().optional(),
        endTime: z
            .string()
            .datetime({ offset: true })
            .transform((v) => new Date(v))
            .optional(),
        groupBy: z.number().int().optional(),
        insertionDate: z.boolean().optional(),
        ledger: z.string(),
        pageSize: z.number().int().optional(),
        query: z.record(z.any()).optional(),
        startTime: z
            .string()
            .datetime({ offset: true })
            .transform((v) => new Date(v))
            .optional(),
    });
    V2GetVolumesWithBalancesRequest$.outboundSchema = z.object({
        cursor: z.string().optional(),
        endTime: z
            .date()
            .transform((v) => v.toISOString())
            .optional(),
        groupBy: z.number().int().optional(),
        insertionDate: z.boolean().optional(),
        ledger: z.string(),
        pageSize: z.number().int().optional(),
        query: z.record(z.any()).optional(),
        startTime: z
            .date()
            .transform((v) => v.toISOString())
            .optional(),
    });
})(V2GetVolumesWithBalancesRequest$ || (exports.V2GetVolumesWithBalancesRequest$ = V2GetVolumesWithBalancesRequest$ = {}));
/** @internal */
var V2GetVolumesWithBalancesResponse$;
(function (V2GetVolumesWithBalancesResponse$) {
    V2GetVolumesWithBalancesResponse$.inboundSchema = z
        .object({
        ContentType: z.string(),
        StatusCode: z.number().int(),
        RawResponse: z.instanceof(Response),
        V2VolumesWithBalanceCursorResponse: shared.V2VolumesWithBalanceCursorResponse$.inboundSchema.optional(),
    })
        .transform((v) => {
        return (0, primitives_1.remap)(v, {
            ContentType: "contentType",
            StatusCode: "statusCode",
            RawResponse: "rawResponse",
            V2VolumesWithBalanceCursorResponse: "v2VolumesWithBalanceCursorResponse",
        });
    });
    V2GetVolumesWithBalancesResponse$.outboundSchema = z
        .object({
        contentType: z.string(),
        statusCode: z.number().int(),
        rawResponse: z.instanceof(Response).transform(() => {
            throw new Error("Response cannot be serialized");
        }),
        v2VolumesWithBalanceCursorResponse: shared.V2VolumesWithBalanceCursorResponse$.outboundSchema.optional(),
    })
        .transform((v) => {
        return (0, primitives_1.remap)(v, {
            contentType: "ContentType",
            statusCode: "StatusCode",
            rawResponse: "RawResponse",
            v2VolumesWithBalanceCursorResponse: "V2VolumesWithBalanceCursorResponse",
        });
    });
})(V2GetVolumesWithBalancesResponse$ || (exports.V2GetVolumesWithBalancesResponse$ = V2GetVolumesWithBalancesResponse$ = {}));
