// @formance/formance-sdk/lib/config.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SDK_METADATA = exports.serverURLFromOptions = exports.ServerList = void 0;
const url_1 = require("./url");
/**
 * Contains the list of servers available to the SDK
 */
exports.ServerList = [
    /**
     * local server
     */
    "http://localhost",
];
function serverURLFromOptions(options) {
    var _a;
    let serverURL = options.serverURL;
    const params = {};
    if (!serverURL) {
        const serverIdx = (_a = options.serverIdx) !== null && _a !== void 0 ? _a : 0;
        if (serverIdx < 0 || serverIdx >= exports.ServerList.length) {
            throw new Error(`Invalid server index ${serverIdx}`);
        }
        serverURL = exports.ServerList[serverIdx] || "";
    }
    const u = (0, url_1.pathToFunc)(serverURL)(params);
    return new URL(u);
}
exports.serverURLFromOptions = serverURLFromOptions;
exports.SDK_METADATA = {
    language: "typescript",
    openapiDocVersion: "v2.0.0-rc.33",
    sdkVersion: "2.4.4",
    genVersion: "2.340.3",
    userAgent: "speakeasy-sdk/typescript 2.4.4 2.340.3 v2.0.0-rc.33 @formance/formance-sdk",
};
